<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-09-29" />

<title>DMD-MYOrganoids</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  { color: #cccccc; background-color: #303030; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ffcfaf; } /* Alert */
code span.an { color: #7f9f7f; font-weight: bold; } /* Annotation */
code span.at { } /* Attribute */
code span.bn { color: #dca3a3; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #f0dfaf; } /* ControlFlow */
code span.ch { color: #dca3a3; } /* Char */
code span.cn { color: #dca3a3; font-weight: bold; } /* Constant */
code span.co { color: #7f9f7f; } /* Comment */
code span.cv { color: #7f9f7f; font-weight: bold; } /* CommentVar */
code span.do { color: #7f9f7f; } /* Documentation */
code span.dt { color: #dfdfbf; } /* DataType */
code span.dv { color: #dcdccc; } /* DecVal */
code span.er { color: #c3bf9f; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #c0bed1; } /* Float */
code span.fu { color: #efef8f; } /* Function */
code span.im { } /* Import */
code span.in { color: #7f9f7f; font-weight: bold; } /* Information */
code span.kw { color: #f0dfaf; } /* Keyword */
code span.op { color: #f0efd0; } /* Operator */
code span.ot { color: #efef8f; } /* Other */
code span.pp { color: #ffcfaf; font-weight: bold; } /* Preprocessor */
code span.sc { color: #dca3a3; } /* SpecialChar */
code span.ss { color: #cc9393; } /* SpecialString */
code span.st { color: #cc9393; } /* String */
code span.va { } /* Variable */
code span.vs { color: #cc9393; } /* VerbatimString */
code span.wa { color: #7f9f7f; font-weight: bold; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">My new website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    About me
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./aboutme.html">Learn more about me</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Project
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./DMD-MYOrganoids.html">DMD-MYOrganoids</a>
    </li>
    <li>
      <a href="./project2.html">Project part 2</a>
    </li>
  </ul>
</li>
<li>
  <a href="./contact.html">Contact</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/meanwhy22/my-first-website">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DMD-MYOrganoids</h1>
<h4 class="date">2025-09-29</h4>

</div>


<div id="data-preprocessing" class="section level2">
<h2>DATA PREPROCESSING</h2>
<p>In this practice I’m going to note down all the steps to re-analyze a
simple RNA-seq study. Data are obtained from our collaborator (Genethon;
<a href="https://www.biorxiv.org/content/10.1101/2023.07.26.550063v4"
class="uri">https://www.biorxiv.org/content/10.1101/2023.07.26.550063v4</a>)
who generated an organoid-like structure from DMD muscle cells (derived
from patient iPSCs) co-cultured with fibroblasts. This dataset contains
<strong>12 samples</strong> of <strong>3 conditions</strong>: IsoCTR,
Untreated (disease exacerbated due to fibroblast incorporation), and
Treated (MYOrganoid treated with high dose of AAV gene therapy). Each
condition includes <strong>4 biological replicates</strong>.</p>
<div id="quality-control-of-raw-sequencing-reads"
class="section level3">
<h3>Quality Control of Raw Sequencing Reads</h3>
<p>First, raw fastq files are imported into FastQC for some quality
control checks. In return, we receive an HTML-based report on each fastq
file.</p>
<p>It seems that the adapters haven’t been cut. We’ll use
<code>fastp</code>, an all-in-one fastq file preprocessor, for this
job.</p>
<p><u>Note:</u> <em>If the job keeps crashing, try to do half the number
of samples until sorted BAM, delete fastq files, then proceed the other
half.</em></p>
<p>Often check folder size:</p>
<div id="read-mapping-to-the-reference-genome" class="section level4">
<h4>Read Mapping to the Reference Genome</h4>
<p>Prior to alignment, we need to download human reference genome
(GRCh38/hg38, as done in the preprint) and index it.</p>
<p>Now we’ll map the (paired) reads to the reference. Remember to use
‘clean’ fastq files. To avoid crashing the alignment, <u>directly output
sorted BAM files</u> (skip SAM and unsorted BAM).</p>
</div>
<div id="sambam-file-manipulation-with-samtools" class="section level4">
<h4>SAM/BAM File Manipulation with Samtools</h4>
<ul>
<li><p>Convert SAM to BAM file (binary format) → save space
(DONE)</p></li>
<li><p>BAM files need to be <u>sorted</u> (DONE) and
<u>indexed</u></p></li>
</ul>
</div>
<div id="generate-count-table" class="section level4">
<h4>Generate count table</h4>
<p>Now we pass all the sorted, indexed BAM files through
<code>featureCounts</code> function (subread package) to generate the
count matrix. We also need an annotation file to do this. Parameters
<code>-t "exon" -g gene_id</code> are to count reads mapped to exon then
summarize to gene-level (i.e. many exons belong to one gene id).</p>
<p>Now we can read the count table into R for differential expression
analysis:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>count_tab <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">file =</span> <span class="st">&quot;count_matrix.counts&quot;</span>,</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>                        <span class="at">row.names =</span> <span class="dv">1</span>,</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>                        <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>                        <span class="at">comment.char =</span> <span class="st">&quot;#&quot;</span>)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co"># Clean column names</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="fu">colnames</span>(count_tab) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^.*</span><span class="sc">\\</span><span class="st">/&quot;</span>, <span class="st">&quot;&quot;</span>, </span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>                         <span class="fu">colnames</span>(count_tab))    <span class="co"># to match everything from the start of the string up to and including the last slash /</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="fu">colnames</span>(count_tab) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.sorted</span><span class="sc">\\</span><span class="st">.bam$&quot;</span>, <span class="st">&quot;&quot;</span>, </span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>                         <span class="fu">colnames</span>(count_tab))    <span class="co"># drop &quot;.sorted.bam&quot;</span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="fu">colnames</span>(count_tab) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^</span><span class="sc">\\</span><span class="st">.</span><span class="sc">\\</span><span class="st">.reads_mapping</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;&quot;</span>, </span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>                            <span class="fu">colnames</span>(count_tab)) <span class="co"># from the start, drop &quot;..reads_mapping&quot;</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="fu">colnames</span>(count_tab)</span></code></pre></div>
<pre><code>##  [1] &quot;Chr&quot;         &quot;Start&quot;      
##  [3] &quot;End&quot;         &quot;Strand&quot;     
##  [5] &quot;Length&quot;      &quot;SRR32076135&quot;
##  [7] &quot;SRR32076136&quot; &quot;SRR32076137&quot;
##  [9] &quot;SRR32076138&quot; &quot;SRR32076139&quot;
## [11] &quot;SRR32076140&quot; &quot;SRR32076141&quot;
## [13] &quot;SRR32076142&quot; &quot;SRR32076143&quot;
## [15] &quot;SRR32076144&quot; &quot;SRR32076145&quot;
## [17] &quot;SRR32076146&quot;</code></pre>
<p>Drop all annotation columns and keep the counts only:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>count_tab <span class="ot">&lt;-</span> count_tab[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">head</span>(count_tab)</span></code></pre></div>
<pre><code>##                 SRR32076135 SRR32076136
## ENSG00000186092           0           0
## ENSG00000284733           0           0
## ENSG00000284662           0           0
## ENSG00000187634         414         289
## ENSG00000188976        2342        2056
## ENSG00000187961         627         557
##                 SRR32076137 SRR32076138
## ENSG00000186092           0           0
## ENSG00000284733           0           0
## ENSG00000284662           0           0
## ENSG00000187634         107         437
## ENSG00000188976        1091        2672
## ENSG00000187961         259         643
##                 SRR32076139 SRR32076140
## ENSG00000186092           0           0
## ENSG00000284733           0           0
## ENSG00000284662           0           0
## ENSG00000187634         242         375
## ENSG00000188976        1551        2108
## ENSG00000187961         361         502
##                 SRR32076141 SRR32076142
## ENSG00000186092           0           0
## ENSG00000284733           0           0
## ENSG00000284662           0           0
## ENSG00000187634         337         138
## ENSG00000188976        2394        3062
## ENSG00000187961         642         563
##                 SRR32076143 SRR32076144
## ENSG00000186092           0           0
## ENSG00000284733           0           0
## ENSG00000284662           0           0
## ENSG00000187634         419         345
## ENSG00000188976        2306        2449
## ENSG00000187961         594         559
##                 SRR32076145 SRR32076146
## ENSG00000186092           0           0
## ENSG00000284733           0           0
## ENSG00000284662           0           0
## ENSG00000187634         117         111
## ENSG00000188976        1196        2706
## ENSG00000187961         278         495</code></pre>
<p>Also read the <strong>metadata</strong> file:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;metadata.xlsx&quot;</span>)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(metadata[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)]) <span class="co"># keep 2 columns: sample and sample_type</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>meta<span class="sc">$</span>sample_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(meta<span class="sc">$</span>sample_type) <span class="co"># sample_type as factor</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co"># sample names have to be rownames</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="fu">rownames</span>(meta) <span class="ot">&lt;-</span> meta<span class="sc">$</span>sample</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>meta<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co"># Map row names (sample names) from meta to column names of count table</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>name_map <span class="ot">&lt;-</span> <span class="fu">setNames</span>(metadata<span class="sc">$</span>sample, metadata<span class="sc">$</span>sample_id)</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="fu">colnames</span>(count_tab) <span class="ot">&lt;-</span> name_map[<span class="fu">colnames</span>(count_tab)]</span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="co"># Reorder meta to match the order of columns in count table</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>meta <span class="ot">&lt;-</span> meta[<span class="fu">colnames</span>(count_tab), , drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a><span class="co"># Check if it&#39;s good to go</span></span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(meta) <span class="sc">==</span> <span class="fu">colnames</span>(count_tab))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
</div>
</div>
</div>
<div id="deseq2-analysis" class="section level2">
<h2>DESeq2 ANALYSIS</h2>
<div id="create-deseqdataset-object" class="section level4">
<h4>Create DESeqDataSet object</h4>
<p>DESeq2 expects a raw count matrix as input. The model will internally
correct for library size.</p>
<p><u>Note:</u> <code>nrow(colData) == ncol(countData)</code></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  <span class="at">countData =</span> count_tab, </span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>  <span class="at">colData =</span> meta,</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> sample_type)</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>dds</span></code></pre></div>
<pre><code>## class: DESeqDataSet 
## dim: 20420 12 
## metadata(1): version
## assays(1): counts
## rownames(20420): ENSG00000186092
##   ENSG00000284733 ... ENSG00000277475
##   ENSG00000268674
## rowData names(0):
## colnames(12): DMDdEx8-9-high-1
##   DMDdEx8-9-high-2 ... corrDMDdEx8-9-2
##   corrDMDdEx8-9-1
## colData names(1): sample_type</code></pre>
</div>
<div id="filter-low-count-genes" class="section level4">
<h4>Filter low count genes</h4>
<p>Here we’re keeping only genes that have at least 10 read counts in at
least 4 samples. In the mean-sd plot, the variance is stabilized,
meaning <u>genes with low and high mean counts all have similar
variances</u>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>dds_filt <span class="ot">&lt;-</span> dds[keep,]</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="fu">library</span>(vsn)</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_filt)</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="fu">meanSdPlot</span>(<span class="fu">assay</span>(vsd))</span></code></pre></div>
<p><img src="DMD-MYOrganoids_files/figure-html/filtering-1.png" width="672" /></p>
</div>
<div id="data-quality-control-by-clustering-samples"
class="section level4">
<h4>Data quality control by clustering samples</h4>
<p>For differential expression analysis, DESeq2 operates directly on the
<em>raw counts</em>. However, for clustering and visualization, we
better use normalized and transformed counts. DESeq2 uses a
normalization method called <strong>median of ratios</strong>.</p>
<p><u>Note:</u> we have to run differential expression analysis to add
normalization factors (since it’s an all-in-one function)</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co"># DESeqDataSet object does not store a matrix of normalized counts by default</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="co"># We can to get it by: counts(dds, normalized=TRUE)  </span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>dds_filt <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_filt)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>select_top_genes <span class="ot">&lt;-</span> <span class="fu">order</span>(<span class="fu">rowMeans</span>(<span class="fu">counts</span>(dds_filt, <span class="at">normalized=</span><span class="cn">TRUE</span>)), <span class="co"># mean of normalized counts of each gene across samples                 </span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>                                     <span class="at">decreasing=</span><span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>] <span class="co"># select top 50 genes with highest mean expression  </span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_filt)[, <span class="st">&quot;sample_type&quot;</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])  </span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a><span class="co"># log2 transformation of normalized counts </span></span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a><span class="co"># pc = 1 is pseudocount ==&gt; log2(n + 1) </span></span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a>ntd <span class="ot">&lt;-</span> <span class="fu">normTransform</span>(dds_filt, <span class="at">f =</span> log2, <span class="at">pc =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>This is a <strong>heatmap</strong> of <u>top 50 highly expressed
genes</u>. Expression values are presented as normalized,
log2-transformed counts and are row-scaled (so that mean expression of
each gene across samples is 0). Here we see samples are clustered based
on top gene expression.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;pheatmap&quot;</span>)</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="fu">assay</span>(ntd)[select_top_genes,],          </span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">&quot;row&quot;</span>,     <span class="co"># z score     </span></span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>         <span class="at">cluster_rows=</span><span class="cn">TRUE</span>,           </span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>         <span class="at">show_rownames=</span><span class="cn">TRUE</span>,                    </span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>         <span class="at">cluster_cols=</span><span class="cn">TRUE</span>,           </span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a>         <span class="at">annotation_col=</span>df)</span></code></pre></div>
<p><img src="DMD-MYOrganoids_files/figure-html/unnamed-chunk-15-1.png" width="768" /></p>
<p>We can make a heatmap of (euclidean) <strong>distances</strong>
between each pair of samples. With this, we have an overall idea of
<u>how samples are similar to each other</u>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>sampleDists <span class="ot">&lt;-</span> <span class="fu">dist</span>(<span class="fu">t</span>(<span class="fu">assay</span>(vsd)))  </span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;RColorBrewer&quot;</span>) </span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>sampleDistMatrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(sampleDists)</span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>( <span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Blues&quot;</span>)) )(<span class="dv">255</span>) </span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a><span class="fu">pheatmap</span>(sampleDistMatrix,          </span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>         <span class="at">clustering_distance_rows=</span>sampleDists,    </span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a>         <span class="at">clustering_distance_cols=</span>sampleDists,          </span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a>         <span class="at">col=</span>colors)</span></code></pre></div>
<p><img src="DMD-MYOrganoids_files/figure-html/unnamed-chunk-16-1.png" width="768" /></p>
<p>Another way to evaluate the <strong>batch effects</strong> is to plot
the principle components of samples (<strong>PCA</strong>).</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup=</span><span class="st">&quot;sample_type&quot;</span>,</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>        <span class="at">ntop=</span><span class="dv">10000</span>)</span></code></pre></div>
<pre><code>## using ntop=10000 top features by variance</code></pre>
<p><img src="DMD-MYOrganoids_files/figure-html/unnamed-chunk-17-1.png" width="480" /></p>
</div>
<div id="differential-expression-analysis" class="section level4">
<h4>Differential expression analysis</h4>
<p><code>DESeq(dds)</code> performs:</p>
<ul>
<li><p>estimation of size factors</p></li>
<li><p>estimation of dispersion</p></li>
<li><p>Negative Binomial GLM fitting and Wald statistics</p></li>
</ul>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="co"># If needed, set &#39;IsoCTR&#39; as baseline condition :</span></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a><span class="co"># dds_filt$sample_type &lt;- relevel(dds_filt$sample_type, ref = &quot;IsoCTR&quot;)</span></span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a><span class="co"># dds_filt &lt;- DESeq(dds_filt)</span></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a><span class="fu">resultsNames</span>(dds_filt) <span class="co"># names of the individual effects (coefficients)</span></span></code></pre></div>
<pre><code>## [1] &quot;Intercept&quot;                      
## [2] &quot;sample_type_Treated_vs_IsoCTR&quot;  
## [3] &quot;sample_type_Untreated_vs_IsoCTR&quot;</code></pre>
<p>p values are corrected by Benjamin-Hochberg with
<code>alpha = 0.05</code>. <em>How many genes were (significantly)
up/down-regulated in</em> ‘Treated’ <em>compared to</em>
‘IsoCTR’<em>?</em></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>Treated_vs_IsoCTR <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_filt,</span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>                             <span class="at">name =</span> <span class="st">&quot;sample_type_Treated_vs_IsoCTR&quot;</span>,</span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>                             <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a><span class="fu">summary</span>(Treated_vs_IsoCTR)</span></code></pre></div>
<pre><code>## 
## out of 14658 with nonzero total read count
## adjusted p-value &lt; 0.05
## LFC &gt; 0 (up)       : 3771, 26%
## LFC &lt; 0 (down)     : 4262, 29%
## outliers [1]       : 3, 0.02%
## low counts [2]     : 0, 0%
## (mean count &lt; 5)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p>For the <strong>volcano plot</strong>, we’ll use the full
<code>results</code> table and highlight the significant hits. Be
careful that there are genes where <code>padj</code> is missing due to
<u>independent filtering inside DESeq2</u> (prior to multiple testing
correction) or <code>log2FoldChange</code> is missing <u>if a gene has
zero counts across all samples of one condition</u>.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="co"># Load libraries </span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a>Treated_vs_IsoCTR_DEGs <span class="ot">&lt;-</span> Treated_vs_IsoCTR <span class="sc">%&gt;%</span> </span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;ENSEMBL&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a>  <span class="co"># Remove genes where log2FC or padj is missing</span></span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(log2FoldChange), <span class="sc">!</span><span class="fu">is.na</span>(padj))</span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a>Treated_vs_IsoCTR_DEGs<span class="sc">$</span>regulation <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb26-15"><a href="#cb26-15" tabindex="-1"></a>  Treated_vs_IsoCTR_DEGs<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> </span>
<span id="cb26-16"><a href="#cb26-16" tabindex="-1"></a>    Treated_vs_IsoCTR_DEGs<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="st">&quot;Up&quot;</span>,</span>
<span id="cb26-17"><a href="#cb26-17" tabindex="-1"></a>  <span class="fu">ifelse</span>( Treated_vs_IsoCTR_DEGs<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span></span>
<span id="cb26-18"><a href="#cb26-18" tabindex="-1"></a>            Treated_vs_IsoCTR_DEGs<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>, <span class="st">&quot;Down&quot;</span>,</span>
<span id="cb26-19"><a href="#cb26-19" tabindex="-1"></a>          <span class="st">&quot;NotSig&quot;</span>)</span>
<span id="cb26-20"><a href="#cb26-20" tabindex="-1"></a>  )</span>
<span id="cb26-21"><a href="#cb26-21" tabindex="-1"></a>  </span>
<span id="cb26-22"><a href="#cb26-22" tabindex="-1"></a><span class="co"># Count up/down genes</span></span>
<span id="cb26-23"><a href="#cb26-23" tabindex="-1"></a>counts <span class="ot">&lt;-</span> Treated_vs_IsoCTR_DEGs <span class="sc">%&gt;%</span></span>
<span id="cb26-24"><a href="#cb26-24" tabindex="-1"></a>  <span class="fu">filter</span>(regulation <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Up&quot;</span>,<span class="st">&quot;Down&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-25"><a href="#cb26-25" tabindex="-1"></a>  <span class="fu">group_by</span>(regulation) <span class="sc">%&gt;%</span></span>
<span id="cb26-26"><a href="#cb26-26" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-27"><a href="#cb26-27" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> regulation, <span class="at">values_from =</span> n, <span class="at">values_fill =</span> <span class="dv">0</span>)</span>
<span id="cb26-28"><a href="#cb26-28" tabindex="-1"></a></span>
<span id="cb26-29"><a href="#cb26-29" tabindex="-1"></a><span class="co"># Volcano plot</span></span>
<span id="cb26-30"><a href="#cb26-30" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> Treated_vs_IsoCTR_DEGs,</span>
<span id="cb26-31"><a href="#cb26-31" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange,</span>
<span id="cb26-32"><a href="#cb26-32" tabindex="-1"></a>                <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj))) <span class="sc">+</span></span>
<span id="cb26-33"><a href="#cb26-33" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">col =</span> regulation)) <span class="sc">+</span></span>
<span id="cb26-34"><a href="#cb26-34" tabindex="-1"></a>  </span>
<span id="cb26-35"><a href="#cb26-35" tabindex="-1"></a>  <span class="co"># Add threshold lines</span></span>
<span id="cb26-36"><a href="#cb26-36" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), </span>
<span id="cb26-37"><a href="#cb26-37" tabindex="-1"></a>             <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, </span>
<span id="cb26-38"><a href="#cb26-38" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>) <span class="sc">+</span></span>
<span id="cb26-39"><a href="#cb26-39" tabindex="-1"></a>  </span>
<span id="cb26-40"><a href="#cb26-40" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>),</span>
<span id="cb26-41"><a href="#cb26-41" tabindex="-1"></a>           <span class="at">col =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb26-42"><a href="#cb26-42" tabindex="-1"></a>           <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>) <span class="sc">+</span></span>
<span id="cb26-43"><a href="#cb26-43" tabindex="-1"></a>  </span>
<span id="cb26-44"><a href="#cb26-44" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Down&quot;</span><span class="ot">=</span><span class="st">&quot;#00AFBB&quot;</span>,</span>
<span id="cb26-45"><a href="#cb26-45" tabindex="-1"></a>                                <span class="st">&quot;NotSig&quot;</span><span class="ot">=</span><span class="st">&quot;grey&quot;</span>,</span>
<span id="cb26-46"><a href="#cb26-46" tabindex="-1"></a>                                <span class="st">&quot;Up&quot;</span><span class="ot">=</span><span class="st">&quot;#E41A1C&quot;</span>),</span>
<span id="cb26-47"><a href="#cb26-47" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Down&quot;</span><span class="ot">=</span><span class="st">&quot;Downregulated&quot;</span>, </span>
<span id="cb26-48"><a href="#cb26-48" tabindex="-1"></a>                                <span class="st">&quot;NotSig&quot;</span><span class="ot">=</span><span class="st">&quot;Not significant&quot;</span>, </span>
<span id="cb26-49"><a href="#cb26-49" tabindex="-1"></a>                                <span class="st">&quot;Up&quot;</span><span class="ot">=</span><span class="st">&quot;Upregulated&quot;</span>)) <span class="sc">+</span></span>
<span id="cb26-50"><a href="#cb26-50" tabindex="-1"></a>  </span>
<span id="cb26-51"><a href="#cb26-51" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;log2FC&quot;</span>,</span>
<span id="cb26-52"><a href="#cb26-52" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;-log10(padj)&quot;</span>,</span>
<span id="cb26-53"><a href="#cb26-53" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">&quot;Regulation&quot;</span>, </span>
<span id="cb26-54"><a href="#cb26-54" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;Treated vs IsoCTR&quot;</span>) <span class="sc">+</span> </span>
<span id="cb26-55"><a href="#cb26-55" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb26-56"><a href="#cb26-56" tabindex="-1"></a>  </span>
<span id="cb26-57"><a href="#cb26-57" tabindex="-1"></a>  <span class="co"># Annotate number of up/down genes on the plot </span></span>
<span id="cb26-58"><a href="#cb26-58" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> <span class="fu">max</span>(Treated_vs_IsoCTR_DEGs<span class="sc">$</span>log2FoldChange, <span class="at">na.rm=</span><span class="cn">TRUE</span>), </span>
<span id="cb26-59"><a href="#cb26-59" tabindex="-1"></a>           <span class="at">y =</span> <span class="fu">max</span>(<span class="sc">-</span><span class="fu">log10</span>(Treated_vs_IsoCTR_DEGs<span class="sc">$</span>padj), <span class="at">na.rm=</span><span class="cn">TRUE</span>), </span>
<span id="cb26-60"><a href="#cb26-60" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;Up: &quot;</span>, counts<span class="sc">$</span>Up), </span>
<span id="cb26-61"><a href="#cb26-61" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">&quot;#E41A1C&quot;</span>) <span class="sc">+</span></span>
<span id="cb26-62"><a href="#cb26-62" tabindex="-1"></a>  </span>
<span id="cb26-63"><a href="#cb26-63" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> <span class="fu">min</span>(Treated_vs_IsoCTR_DEGs<span class="sc">$</span>log2FoldChange, <span class="at">na.rm=</span><span class="cn">TRUE</span>), </span>
<span id="cb26-64"><a href="#cb26-64" tabindex="-1"></a>           <span class="at">y =</span> <span class="fu">max</span>(<span class="sc">-</span><span class="fu">log10</span>(Treated_vs_IsoCTR_DEGs<span class="sc">$</span>padj), <span class="at">na.rm=</span><span class="cn">TRUE</span>), </span>
<span id="cb26-65"><a href="#cb26-65" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;Down: &quot;</span>, counts<span class="sc">$</span>Down), </span>
<span id="cb26-66"><a href="#cb26-66" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">&quot;#00AFBB&quot;</span>)</span>
<span id="cb26-67"><a href="#cb26-67" tabindex="-1"></a>p</span></code></pre></div>
<p><img src="DMD-MYOrganoids_files/figure-html/volcano_plot-1.png" width="672" /></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a><span class="co"># Write the file</span></span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a><span class="fu">write.table</span>(Treated_vs_IsoCTR_DEGs,</span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">&quot;DEG_List/Treated_vs_IsoCTR_DEGs.txt&quot;</span>,</span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb27-6"><a href="#cb27-6" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><em>How many genes were (significantly) up/down-regulated in</em>
‘Untreated’ <em>compared to</em> ‘IsoCTR’<em>?</em></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>Untreated_vs_IsoCTR <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_filt,</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>                               <span class="at">name =</span> <span class="st">&quot;sample_type_Untreated_vs_IsoCTR&quot;</span>,</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a>                               <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a><span class="fu">summary</span>(Untreated_vs_IsoCTR)</span></code></pre></div>
<pre><code>## 
## out of 14658 with nonzero total read count
## adjusted p-value &lt; 0.05
## LFC &gt; 0 (up)       : 1933, 13%
## LFC &lt; 0 (down)     : 2685, 18%
## outliers [1]       : 3, 0.02%
## low counts [2]     : 0, 0%
## (mean count &lt; 5)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="co"># Remove genes where log2FC or padj is missing</span></span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a><span class="co"># Make row names a column EMSEMBL</span></span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>Untreated_vs_IsoCTR_DEGs <span class="ot">&lt;-</span> </span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a>  Untreated_vs_IsoCTR <span class="sc">%&gt;%</span></span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb30-7"><a href="#cb30-7" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(log2FoldChange), <span class="sc">!</span><span class="fu">is.na</span>(padj)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-8"><a href="#cb30-8" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;ENSEMBL&quot;</span>)</span>
<span id="cb30-9"><a href="#cb30-9" tabindex="-1"></a></span>
<span id="cb30-10"><a href="#cb30-10" tabindex="-1"></a><span class="co"># Add column &#39;regulation&#39;</span></span>
<span id="cb30-11"><a href="#cb30-11" tabindex="-1"></a>Untreated_vs_IsoCTR_DEGs<span class="sc">$</span>regulation <span class="ot">&lt;-</span></span>
<span id="cb30-12"><a href="#cb30-12" tabindex="-1"></a><span class="fu">ifelse</span>( Untreated_vs_IsoCTR_DEGs<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span></span>
<span id="cb30-13"><a href="#cb30-13" tabindex="-1"></a>          Untreated_vs_IsoCTR_DEGs<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="st">&quot;Up&quot;</span>,</span>
<span id="cb30-14"><a href="#cb30-14" tabindex="-1"></a>  <span class="fu">ifelse</span>( Untreated_vs_IsoCTR_DEGs<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span></span>
<span id="cb30-15"><a href="#cb30-15" tabindex="-1"></a>            Untreated_vs_IsoCTR_DEGs<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>, <span class="st">&quot;Down&quot;</span>,</span>
<span id="cb30-16"><a href="#cb30-16" tabindex="-1"></a>          <span class="st">&quot;NotSig&quot;</span>)</span>
<span id="cb30-17"><a href="#cb30-17" tabindex="-1"></a>)</span>
<span id="cb30-18"><a href="#cb30-18" tabindex="-1"></a></span>
<span id="cb30-19"><a href="#cb30-19" tabindex="-1"></a><span class="co"># Write the file</span></span>
<span id="cb30-20"><a href="#cb30-20" tabindex="-1"></a><span class="fu">write.table</span>(Untreated_vs_IsoCTR_DEGs,</span>
<span id="cb30-21"><a href="#cb30-21" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">&quot;DEG_List/Untreated_vs_IsoCTR_DEGs.txt&quot;</span>,</span>
<span id="cb30-22"><a href="#cb30-22" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb30-23"><a href="#cb30-23" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb30-24"><a href="#cb30-24" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><em>How many genes were (significantly) up/down-regulated in</em>
‘Treated’ <em>compared to</em> ‘Untreated’<em>?</em></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>Treated_vs_Untreated <span class="ot">&lt;-</span> </span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>  <span class="fu">results</span>(dds_filt, </span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a>          <span class="at">contrast=</span><span class="fu">c</span>(<span class="st">&quot;sample_type&quot;</span>,<span class="st">&quot;Treated&quot;</span>,<span class="st">&quot;Untreated&quot;</span>),</span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a>          <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a><span class="fu">summary</span>(Treated_vs_Untreated)</span></code></pre></div>
<pre><code>## 
## out of 14658 with nonzero total read count
## adjusted p-value &lt; 0.05
## LFC &gt; 0 (up)       : 315, 2.1%
## LFC &lt; 0 (down)     : 389, 2.7%
## outliers [1]       : 3, 0.02%
## low counts [2]     : 285, 1.9%
## (mean count &lt; 9)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
</div>
<div id="crossing-de-gene-sets-venn-diagram" class="section level4">
<h4>Crossing DE gene sets: Venn diagram</h4>
<p>We wanna intersect 3 sets of DE genes:</p>
<ul>
<li><p>Untreated vs IsoCTR: genes that are differentially expressed in
DMD organoid compared to isogenic control</p></li>
<li><p>Treated vs Untreated: genes that are differentially expressed in
AAV-treated DMD organoid compared to non-treated (presumably ‘corrected’
by gene therapy)</p></li>
<li><p>Treated vs IsoCTR: presumably ‘not corrected’ by gene
therapy</p></li>
</ul>
<p>We can use <code>abs(log2FoldChange) &gt; 1</code> but it seems too
strict.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>  <span class="fu">rownames</span>(Untreated_vs_IsoCTR <span class="sc">%&gt;%</span> </span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>             <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>             <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">0</span>)),</span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a>  <span class="fu">rownames</span>(Treated_vs_Untreated <span class="sc">%&gt;%</span> </span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a>             <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb33-7"><a href="#cb33-7" tabindex="-1"></a>             <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">0</span>)),</span>
<span id="cb33-8"><a href="#cb33-8" tabindex="-1"></a>  <span class="fu">rownames</span>(Treated_vs_IsoCTR <span class="sc">%&gt;%</span> </span>
<span id="cb33-9"><a href="#cb33-9" tabindex="-1"></a>             <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb33-10"><a href="#cb33-10" tabindex="-1"></a>             <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb33-11"><a href="#cb33-11" tabindex="-1"></a>)</span>
<span id="cb33-12"><a href="#cb33-12" tabindex="-1"></a></span>
<span id="cb33-13"><a href="#cb33-13" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggVennDiagram&quot;</span>)</span>
<span id="cb33-14"><a href="#cb33-14" tabindex="-1"></a><span class="fu">ggVennDiagram</span>(x, <span class="at">label_alpha =</span> <span class="dv">0</span>,</span>
<span id="cb33-15"><a href="#cb33-15" tabindex="-1"></a>              <span class="at">label =</span> <span class="st">&quot;count&quot;</span>,</span>
<span id="cb33-16"><a href="#cb33-16" tabindex="-1"></a>              <span class="at">category.names =</span> <span class="fu">c</span>(<span class="st">&quot;Untreated</span><span class="sc">\n</span><span class="st">vs IsoCTR&quot;</span>,</span>
<span id="cb33-17"><a href="#cb33-17" tabindex="-1"></a>                                 <span class="st">&quot;Treated</span><span class="sc">\n</span><span class="st">vs Untreated&quot;</span>,</span>
<span id="cb33-18"><a href="#cb33-18" tabindex="-1"></a>                                 <span class="st">&quot;Treated</span><span class="sc">\n</span><span class="st">vs IsoCTR&quot;</span>)) <span class="sc">+</span></span>
<span id="cb33-19"><a href="#cb33-19" tabindex="-1"></a>  </span>
<span id="cb33-20"><a href="#cb33-20" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">4</span>, <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb33-21"><a href="#cb33-21" tabindex="-1"></a>        <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>)) <span class="sc">+</span></span>
<span id="cb33-22"><a href="#cb33-22" tabindex="-1"></a>  </span>
<span id="cb33-23"><a href="#cb33-23" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;white&quot;</span>, <span class="at">high =</span> <span class="st">&quot;orange&quot;</span>)</span></code></pre></div>
<p><img src="DMD-MYOrganoids_files/figure-html/venn-1.png" width="960" /></p>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level4">
<h4>Gene Set Enrichment Analysis (GSEA)</h4>
<p>Prior to GSEA, we need to create a ranked list of the genes. But
first of all, let’s tidy our table of DEGs (just as before).</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="co"># Remove genes where log2FC or padj is missing</span></span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a><span class="co"># Make row names a column EMSEMBL</span></span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a></span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>Treated_vs_Untreated_DEGs <span class="ot">&lt;-</span> </span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a>  Treated_vs_Untreated <span class="sc">%&gt;%</span></span>
<span id="cb34-6"><a href="#cb34-6" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb34-7"><a href="#cb34-7" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(log2FoldChange), <span class="sc">!</span><span class="fu">is.na</span>(padj)) <span class="sc">%&gt;%</span> </span>
<span id="cb34-8"><a href="#cb34-8" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;ENSEMBL&quot;</span>)</span>
<span id="cb34-9"><a href="#cb34-9" tabindex="-1"></a></span>
<span id="cb34-10"><a href="#cb34-10" tabindex="-1"></a><span class="co"># Add column &#39;regulation&#39;</span></span>
<span id="cb34-11"><a href="#cb34-11" tabindex="-1"></a>Treated_vs_Untreated_DEGs<span class="sc">$</span>regulation <span class="ot">&lt;-</span></span>
<span id="cb34-12"><a href="#cb34-12" tabindex="-1"></a><span class="fu">ifelse</span>( Treated_vs_Untreated_DEGs<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span></span>
<span id="cb34-13"><a href="#cb34-13" tabindex="-1"></a>          Treated_vs_Untreated_DEGs<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="st">&quot;Up&quot;</span>,</span>
<span id="cb34-14"><a href="#cb34-14" tabindex="-1"></a>  <span class="fu">ifelse</span>( Treated_vs_Untreated_DEGs<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span></span>
<span id="cb34-15"><a href="#cb34-15" tabindex="-1"></a>            Treated_vs_Untreated_DEGs<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>, <span class="st">&quot;Down&quot;</span>,</span>
<span id="cb34-16"><a href="#cb34-16" tabindex="-1"></a>          <span class="st">&quot;NotSig&quot;</span>)</span>
<span id="cb34-17"><a href="#cb34-17" tabindex="-1"></a>)</span>
<span id="cb34-18"><a href="#cb34-18" tabindex="-1"></a></span>
<span id="cb34-19"><a href="#cb34-19" tabindex="-1"></a><span class="co"># Write the file</span></span>
<span id="cb34-20"><a href="#cb34-20" tabindex="-1"></a><span class="fu">write.table</span>(Treated_vs_Untreated_DEGs,</span>
<span id="cb34-21"><a href="#cb34-21" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">&quot;DEG_List/Treated_vs_Untreated_DEGs.txt&quot;</span>,</span>
<span id="cb34-22"><a href="#cb34-22" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb34-23"><a href="#cb34-23" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb34-24"><a href="#cb34-24" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p>A bunch of <strong>ranking metrics</strong> can be used to rank the
list of input genes. In the end, they should all correlate to the two
phenotypes signatured by the gene set that we’re interested in.</p>
<p>Here we’ll try using the (preferably shrunken) log fold change (LFC)
in DESeq2. Why shrinkage of LFC estimates (toward 0)? - to avoid
inflation by genes with small counts/large standard errors.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>Treat_vs_Untreated_DEGs <span class="ot">&lt;-</span></span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a><span class="fu">read.table</span>(<span class="at">file =</span> <span class="st">&quot;DEG_List/Treated_vs_Untreated_DEGs.txt&quot;</span>,</span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>           <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb35-4"><a href="#cb35-4" tabindex="-1"></a></span>
<span id="cb35-5"><a href="#cb35-5" tabindex="-1"></a>Treat_vs_Untreated_DEGs_ranked <span class="ot">&lt;-</span> Treat_vs_Untreated_DEGs <span class="sc">%&gt;%</span> </span>
<span id="cb35-6"><a href="#cb35-6" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(log2FoldChange)) <span class="sc">%&gt;%</span></span>
<span id="cb35-7"><a href="#cb35-7" tabindex="-1"></a>  <span class="fu">pull</span>(log2FoldChange, <span class="at">name=</span>ENSEMBL) <span class="co"># create named vector</span></span>
<span id="cb35-8"><a href="#cb35-8" tabindex="-1"></a></span>
<span id="cb35-9"><a href="#cb35-9" tabindex="-1"></a><span class="fu">head</span>(Treat_vs_Untreated_DEGs_ranked)</span></code></pre></div>
<pre><code>## ENSG00000166450 ENSG00000169840 
##        1.739995        1.496896 
## ENSG00000105810 ENSG00000274276 
##        1.495937        1.470676 
## ENSG00000187553 ENSG00000163273 
##        1.470314        1.364892</code></pre>
<p>To obtain the <strong>gene set</strong> of interest:</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a>GOBP_all <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(</span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a>  <span class="at">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>,</span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a>  <span class="at">collection =</span> <span class="st">&quot;C5&quot;</span>,</span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a>  <span class="at">subcollection =</span> <span class="st">&quot;GO:BP&quot;</span></span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gs_name, ensembl_gene)</span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a><span class="fu">head</span>(GOBP_all)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 2
##   gs_name                       ensembl_gene
##   &lt;chr&gt;                         &lt;chr&gt;       
## 1 GOBP_10_FORMYLTETRAHYDROFOLA… ENSG0000014…
## 2 GOBP_10_FORMYLTETRAHYDROFOLA… ENSG0000014…
## 3 GOBP_10_FORMYLTETRAHYDROFOLA… ENSG0000013…
## 4 GOBP_10_FORMYLTETRAHYDROFOLA… ENSG0000010…
## 5 GOBP_10_FORMYLTETRAHYDROFOLA… ENSG0000012…
## 6 GOBP_2FE_2S_CLUSTER_ASSEMBLY  ENSG0000018…</code></pre>
<p>Running GSEA with all <a href="GO:BP" class="uri">GO:BP</a> gene
sets:</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a>gsea_GOBP_all <span class="ot">&lt;-</span> <span class="fu">GSEA</span>(</span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a>  Treat_vs_Untreated_DEGs_ranked,</span>
<span id="cb39-4"><a href="#cb39-4" tabindex="-1"></a>  <span class="at">TERM2GENE =</span> GOBP_all,</span>
<span id="cb39-5"><a href="#cb39-5" tabindex="-1"></a>  <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## using &#39;fgsea&#39; for GSEA analysis, please cite Korotkevich et al (2019).</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## Warning in fgseaMultilevel(pathways =
## pathways, stats = stats, minSize = minSize,
## : For some pathways, in reality P-values
## are less than 1e-10. You can set the `eps`
## argument to zero for better estimation.</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="co"># List of gene sets with significant enrichment</span></span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a><span class="fu">head</span>(gsea_GOBP_all<span class="sc">@</span>result, <span class="dv">20</span>)<span class="sc">$</span>ID</span></code></pre></div>
<pre><code>##  [1] &quot;GOBP_MITOTIC_SISTER_CHROMATID_SEGREGATION&quot;                 
##  [2] &quot;GOBP_NEGATIVE_REGULATION_OF_CHROMOSOME_ORGANIZATION&quot;       
##  [3] &quot;GOBP_MITOTIC_CELL_CYCLE_CHECKPOINT_SIGNALING&quot;              
##  [4] &quot;GOBP_SISTER_CHROMATID_SEGREGATION&quot;                         
##  [5] &quot;GOBP_DEFENSE_RESPONSE_TO_BACTERIUM&quot;                        
##  [6] &quot;GOBP_REGULATION_OF_MITOTIC_SISTER_CHROMATID_SEGREGATION&quot;   
##  [7] &quot;GOBP_CHROMOSOME_SEPARATION&quot;                                
##  [8] &quot;GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION&quot;                 
##  [9] &quot;GOBP_MITOTIC_SISTER_CHROMATID_SEPARATION&quot;                  
## [10] &quot;GOBP_CHROMOSOME_SEGREGATION&quot;                               
## [11] &quot;GOBP_NUCLEAR_CHROMOSOME_SEGREGATION&quot;                       
## [12] &quot;GOBP_DNA_TEMPLATED_DNA_REPLICATION&quot;                        
## [13] &quot;GOBP_CELL_CYCLE_CHECKPOINT_SIGNALING&quot;                      
## [14] &quot;GOBP_ATP_SYNTHESIS_COUPLED_ELECTRON_TRANSPORT&quot;             
## [15] &quot;GOBP_DNA_REPLICATION&quot;                                      
## [16] &quot;GOBP_TUMOR_NECROSIS_FACTOR_SUPERFAMILY_CYTOKINE_PRODUCTION&quot;
## [17] &quot;GOBP_OXIDATIVE_PHOSPHORYLATION&quot;                            
## [18] &quot;GOBP_REGULATION_OF_CHROMOSOME_ORGANIZATION&quot;                
## [19] &quot;GOBP_DEFENSE_RESPONSE_TO_VIRUS&quot;                            
## [20] &quot;GOBP_RESPONSE_TO_BACTERIUM&quot;</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb48-2"><a href="#cb48-2" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb48-3"><a href="#cb48-3" tabindex="-1"></a></span>
<span id="cb48-4"><a href="#cb48-4" tabindex="-1"></a><span class="fu">dotplot</span>(gsea_GOBP_all, <span class="at">showCategory =</span> <span class="dv">20</span>,</span>
<span id="cb48-5"><a href="#cb48-5" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">&quot;NES&quot;</span>) <span class="sc">+</span></span>
<span id="cb48-6"><a href="#cb48-6" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Top 20 enriched pathways&quot;</span>)</span></code></pre></div>
<p><img src="DMD-MYOrganoids_files/figure-html/dotplot-1.png" width="768" /></p>
<p><em>What if I wanna see only terms that relate to
“inflammation”?</em></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb49-2"><a href="#cb49-2" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb49-3"><a href="#cb49-3" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" tabindex="-1"></a><span class="co"># Subset result table for terms containing &quot;*inflam*&quot;</span></span>
<span id="cb49-5"><a href="#cb49-5" tabindex="-1"></a>inflam_terms <span class="ot">&lt;-</span> gsea_GOBP_all<span class="sc">@</span>result <span class="sc">%&gt;%</span></span>
<span id="cb49-6"><a href="#cb49-6" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;inflam&quot;</span>, Description, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>))</span>
<span id="cb49-7"><a href="#cb49-7" tabindex="-1"></a></span>
<span id="cb49-8"><a href="#cb49-8" tabindex="-1"></a><span class="co"># Create a new gseaResult object</span></span>
<span id="cb49-9"><a href="#cb49-9" tabindex="-1"></a>gsea_GOBP_inflam <span class="ot">&lt;-</span> gsea_GOBP_all</span>
<span id="cb49-10"><a href="#cb49-10" tabindex="-1"></a><span class="co"># The result table in this new object is the subset one</span></span>
<span id="cb49-11"><a href="#cb49-11" tabindex="-1"></a>gsea_GOBP_inflam<span class="sc">@</span>result <span class="ot">&lt;-</span> inflam_terms</span>
<span id="cb49-12"><a href="#cb49-12" tabindex="-1"></a></span>
<span id="cb49-13"><a href="#cb49-13" tabindex="-1"></a><span class="co"># Plot only &#39;*inflam*&#39;-related sets</span></span>
<span id="cb49-14"><a href="#cb49-14" tabindex="-1"></a><span class="fu">dotplot</span>(gsea_GOBP_inflam, <span class="at">color =</span> <span class="st">&quot;NES&quot;</span>,</span>
<span id="cb49-15"><a href="#cb49-15" tabindex="-1"></a>        <span class="at">showCategory =</span> <span class="fu">nrow</span>(inflam_terms)) <span class="sc">+</span></span>
<span id="cb49-16"><a href="#cb49-16" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Inflammation-related GO terms&quot;</span>)</span></code></pre></div>
<p><img src="DMD-MYOrganoids_files/figure-html/unnamed-chunk-23-1.png" width="768" /></p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb50-2"><a href="#cb50-2" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb50-3"><a href="#cb50-3" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb50-4"><a href="#cb50-4" tabindex="-1"></a></span>
<span id="cb50-5"><a href="#cb50-5" tabindex="-1"></a><span class="co"># Visualization</span></span>
<span id="cb50-6"><a href="#cb50-6" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">gseaplot2</span>(gsea_GOBP_all, <span class="at">geneSetID =</span> <span class="st">&quot;GOBP_POSITIVE_REGULATION_OF_INFLAMMATORY_RESPONSE&quot;</span>,</span>
<span id="cb50-7"><a href="#cb50-7" tabindex="-1"></a>                            <span class="at">pvalue_table =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-8"><a href="#cb50-8" tabindex="-1"></a>p1_list <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(p1, <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb50-9"><a href="#cb50-9" tabindex="-1"></a>                      <span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>),</span>
<span id="cb50-10"><a href="#cb50-10" tabindex="-1"></a>                      <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span>
<span id="cb50-11"><a href="#cb50-11" tabindex="-1"></a></span>
<span id="cb50-12"><a href="#cb50-12" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">gseaplot2</span>(gsea_GOBP_all, <span class="at">geneSetID =</span> <span class="st">&quot;GOBP_MICROTUBULE_ORGANIZING_CENTER_ORGANIZATION&quot;</span>,</span>
<span id="cb50-13"><a href="#cb50-13" tabindex="-1"></a>                            <span class="at">pvalue_table =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-14"><a href="#cb50-14" tabindex="-1"></a>p2_list <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(p2, <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb50-15"><a href="#cb50-15" tabindex="-1"></a>                      <span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>),</span>
<span id="cb50-16"><a href="#cb50-16" tabindex="-1"></a>                      <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span>
<span id="cb50-17"><a href="#cb50-17" tabindex="-1"></a></span>
<span id="cb50-18"><a href="#cb50-18" tabindex="-1"></a><span class="co"># Combine 2 plots</span></span>
<span id="cb50-19"><a href="#cb50-19" tabindex="-1"></a><span class="fu">plot_grid</span>(p1_list, p2_list, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="DMD-MYOrganoids_files/figure-html/gsea-plot-1.png" width="480" /></p>
</div>
<div id="convert-ensembl-id-to-gene-symbol-optional"
class="section level4">
<h4>Convert ENSEMBL ID to gene symbol (optional)</h4>
<p>There are many ways to get gene names based on Ensembl IDs. However,
some IDs, usually corresponding to non-coding RNAs, do not match a gene
symbol. If we care only about protein-coding genes, maybe go back to
<code>featureCounts</code> and consider reads that map CDS.</p>
<ul>
<li><p>Here we’ll try keeping the IDs as they are for those that don’t
have a gene symbol.</p></li>
<li><p>Same for IDs that match multiple gene symbols, we keep the
Ensembl IDs to make it less ambiguous.</p></li>
</ul>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb51-2"><a href="#cb51-2" tabindex="-1"></a>annots <span class="ot">&lt;-</span> <span class="fu">select</span>(org.Hs.eg.db,</span>
<span id="cb51-3"><a href="#cb51-3" tabindex="-1"></a>                 </span>
<span id="cb51-4"><a href="#cb51-4" tabindex="-1"></a>                 <span class="co"># ensembl ids in count table</span></span>
<span id="cb51-5"><a href="#cb51-5" tabindex="-1"></a>                 <span class="at">keys=</span><span class="fu">rownames</span>(count_tab),</span>
<span id="cb51-6"><a href="#cb51-6" tabindex="-1"></a>                 </span>
<span id="cb51-7"><a href="#cb51-7" tabindex="-1"></a>                 <span class="at">columns=</span><span class="fu">c</span>(<span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;GENENAME&quot;</span>), </span>
<span id="cb51-8"><a href="#cb51-8" tabindex="-1"></a>                 <span class="at">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;select()&#39; returned 1:many mapping
## between keys and columns</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" tabindex="-1"></a><span class="co"># Replace missing SYMBOLs with the ENSEMBL ID</span></span>
<span id="cb53-2"><a href="#cb53-2" tabindex="-1"></a>annots<span class="sc">$</span>SYMBOL[<span class="fu">is.na</span>(annots<span class="sc">$</span>SYMBOL)] <span class="ot">&lt;-</span> annots<span class="sc">$</span>ENSEMBL</span></code></pre></div>
<pre><code>## Warning in
## annots$SYMBOL[is.na(annots$SYMBOL)] &lt;-
## annots$ENSEMBL: number of items to replace
## is not a multiple of replacement length</code></pre>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" tabindex="-1"></a><span class="co"># Collapse SYMBOLs, but if &gt;1 SYMBOL, use the ENSEMBL ID instead</span></span>
<span id="cb55-2"><a href="#cb55-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb55-3"><a href="#cb55-3" tabindex="-1"></a>annots <span class="ot">&lt;-</span> annots <span class="sc">%&gt;%</span></span>
<span id="cb55-4"><a href="#cb55-4" tabindex="-1"></a>  <span class="fu">group_by</span>(ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb55-5"><a href="#cb55-5" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb55-6"><a href="#cb55-6" tabindex="-1"></a>    <span class="at">SYMBOL =</span> <span class="fu">ifelse</span>(<span class="fu">n_distinct</span>(SYMBOL) <span class="sc">==</span> <span class="dv">1</span>,</span>
<span id="cb55-7"><a href="#cb55-7" tabindex="-1"></a>                    <span class="fu">unique</span>(SYMBOL),</span>
<span id="cb55-8"><a href="#cb55-8" tabindex="-1"></a>                    ENSEMBL)</span>
<span id="cb55-9"><a href="#cb55-9" tabindex="-1"></a>  )</span>
<span id="cb55-10"><a href="#cb55-10" tabindex="-1"></a></span>
<span id="cb55-11"><a href="#cb55-11" tabindex="-1"></a><span class="co"># Add &#39;SYMBOL&#39; column to DEG table</span></span>
<span id="cb55-12"><a href="#cb55-12" tabindex="-1"></a>Treated_vs_Untreated_DEGs_annot <span class="ot">&lt;-</span> Treated_vs_Untreated_DEGs <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(annots, <span class="at">by =</span> <span class="st">&quot;ENSEMBL&quot;</span>)</span>
<span id="cb55-13"><a href="#cb55-13" tabindex="-1"></a><span class="fu">head</span>(Treated_vs_Untreated_DEGs_annot)</span></code></pre></div>
<pre><code>##           ENSEMBL  baseMean log2FoldChange
## 1 ENSG00000187634  262.6958    -0.02988649
## 2 ENSG00000188976 2060.0851    -0.03827750
## 3 ENSG00000187961  484.3473     0.02745172
## 4 ENSG00000188290 1008.3230    -0.29908219
## 5 ENSG00000187608  255.1310    -1.54775797
## 6 ENSG00000188157 7175.7911     0.04679664
##       lfcSE       stat       pvalue
## 1 0.2634260 -0.1134531 0.9096713783
## 2 0.1034109 -0.3701496 0.7112710455
## 3 0.1223824  0.2243111 0.8225152537
## 4 0.2779788 -1.0759173 0.2819642516
## 5 0.4053978 -3.8178745 0.0001346064
## 6 0.3369123  0.1388986 0.8895303024
##          padj regulation SYMBOL
## 1 0.966504821     NotSig SAMD11
## 2 0.881945752     NotSig  NOC2L
## 3 0.930906475     NotSig KLHL17
## 4 0.645048466     NotSig   HES4
## 5 0.007354729       Down  ISG15
## 6 0.959285164     NotSig   AGRN</code></pre>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
