<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>RNA-seq analysis of DMD muscle organoids</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.0/css/all.min.css">

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style_v2.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Minh's Notebook</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Resume
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="aboutme.html">About Minh</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Projects
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="projects.html">My practice sessions</a>
    </li>
  </ul>
</li>
<li>
  <a href="contact.html">Contact</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://meanwhy22.github.io/my-first-website/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">RNA-seq analysis of DMD muscle
organoids</h1>

</div>


<blockquote>
<p><strong>Date:</strong> 18/10/2025</p>
<p><strong>Description:</strong> <a
href="https://www.biorxiv.org/content/10.1101/2023.07.26.550063v4">Palmieri
et al. (2025)</a> generated an organoid-like structure from DMD muscle
cells (derived from patient iPSCs) co-cultured with fibroblasts. This
dataset contains <strong>12 samples</strong> of <strong>3
conditions</strong>: IsoCTR, Untreated (disease), and Treated (disease
treated with high dose of AAV gene therapy). Each condition includes
<strong>4 biological replicates</strong>. Raw data are retrieved from
the NCBI Bioproject database under accession PRJNA1208956.</p>
</blockquote>
<div id="from-fastq-files-to-count-table" class="section level1">
<h1>From FASTQ files to count table</h1>
<details>
<summary>
<strong>Retrieve fastq files from database</strong>
</summary>
<div class="sourceCode" id="cb1"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Get table from ENA API</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="co"># This table contains the links to all raw fastq files</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-s</span> <span class="st">&quot;https://www.ebi.ac.uk/ena/portal/api/filereport?accession=PRJNA1208956&amp;result=read_run&amp;fields=run_accession,fastq_ftp&quot;</span> <span class="op">&gt;</span> ena_runs.txt</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co"># Download all files</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co"># Paired-end reads ==&gt; 2 fastq files for each sample</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="fu">awk</span> <span class="at">-F</span><span class="st">&#39;\t&#39;</span> <span class="st">&#39;NR&gt;1 {print $2}&#39;</span> ena_runs.txt <span class="kw">|</span> <span class="fu">tr</span> <span class="st">&#39;;&#39;</span> <span class="st">&#39;\n&#39;</span> <span class="kw">|</span> <span class="cf">while</span> <span class="bu">read</span> <span class="va">url</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>    <span class="fu">wget</span> ftp://<span class="va">$url</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
</details>
<details>
<summary>
<strong>Quality control of raw reads</strong>
</summary>
<p>Raw fastq files are imported into <a
href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>
for some quality control checks. In return, we receive an HTML-based
report on each fastq file.</p>
<div class="sourceCode" id="cb2"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="co"># module load fastqc</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="co"># mkdir -p reads_qc</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> fastq_files/<span class="pp">*</span>.fastq.gz</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="va">$sample</span> .fastq.gz<span class="va">)</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>  <span class="ex">fastqc</span> <span class="st">&quot;fastq_files/</span><span class="va">$base</span><span class="st">.fastq.gz&quot;</span> <span class="at">--outdir</span> reads_qc</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>Looks like the adapters haven’t been cut. We’ll use <a
href="https://github.com/OpenGene/fastp"><code>fastp</code></a>, an
all-in-one fastq file preprocessor, for this job.</p>
<div class="sourceCode" id="cb3"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># module load fastp</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co"># mkdir -p clean_reads</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> fastq_files/<span class="pp">*</span>_1.fastq.gz</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>    <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">&quot;</span><span class="va">$sample</span><span class="st">&quot;</span> _1.fastq.gz<span class="va">)</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>    <span class="ex">fastp</span> <span class="dt">\</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>        <span class="at">-i</span> <span class="st">&quot;fastq_files/</span><span class="va">${base}</span><span class="st">_1.fastq.gz&quot;</span> <span class="dt">\</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>        <span class="at">-I</span> <span class="st">&quot;fastq_files/</span><span class="va">${base}</span><span class="st">_2.fastq.gz&quot;</span> <span class="dt">\</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>        <span class="at">-o</span> <span class="st">&quot;clean_reads/</span><span class="va">${base}</span><span class="st">_1_clean.fastq.gz&quot;</span> <span class="dt">\</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>        <span class="at">-O</span> <span class="st">&quot;clean_reads/</span><span class="va">${base}</span><span class="st">_2_clean.fastq.gz&quot;</span> <span class="dt">\</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>        <span class="at">-h</span> <span class="st">&quot;clean_reads/</span><span class="va">${base}</span><span class="st">_fastp.html&quot;</span> <span class="dt">\</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>        <span class="at">-j</span> <span class="st">&quot;clean_reads/</span><span class="va">${base}</span><span class="st">_fastp.json&quot;</span> <span class="dt">\</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>        <span class="at">-w</span> 12     <span class="co"># run one sample at a time using 12 threads</span></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<div class="line-block">Note: If the job keeps crashing, try to do half
the number of samples until sorted BAM, delete fastq files, then proceed
the other half.</div>
<p>Check folder size regularly:</p>
<div class="sourceCode" id="cb4"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">du</span> <span class="at">-sh</span> /shared/home/minhytran/<span class="pp">*</span> <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-h</span></span></code></pre></div>
</details>
<details>
<summary>
<strong>Read mapping to reference genome</strong>
</summary>
<p>Prior to alignment, we need to download human reference genome
(GRCh38/hg38) and index it.</p>
<div class="sourceCode" id="cb5"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co"># Download reference genome</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-o</span> ref_genome/GCF_000001405.40_GRCh38.p14_genomic.fna.gz <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_genomic.fna.gz</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co"># To decompress fasta file :</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co"># gunzip GCF_000001405.40_GRCh38.p14_genomic.fna.gz</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co"># Build genome indexes :</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co"># module load hisat2</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co"># mkdir -p ref_genome/index_hisat2 </span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="ex">hisat2-build</span> ref_genome/Homo_sapiens.GRCh38.dna.primary_assembly.fa <span class="dt">\</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>ref_genome/index_hisat2/human</span></code></pre></div>
<p>Now we’ll map the (paired) reads to the reference. Remember to use
‘clean’ fastq files. To avoid crashing the alignment, directly output
sorted BAM files (skip SAM and unsorted BAM).</p>
<div class="sourceCode" id="cb6"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co"># module load hisat2 samtools</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co"># mkdir -p reads_mapping</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> clean_reads/SRR3207614<span class="pp">*</span>_1_clean.fastq.gz </span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="cf">do</span> </span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>  </span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>  <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">&quot;</span><span class="va">$sample</span><span class="st">&quot;</span> _1_clean.fastq.gz<span class="va">)</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>  </span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>  <span class="ex">hisat2</span> <span class="at">-q</span> <span class="at">-p</span> 12 <span class="dt">\</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>  <span class="at">--summary-file</span> <span class="st">&quot;./reads_mapping/summary_</span><span class="va">${base}</span><span class="st">.txt&quot;</span> <span class="dt">\</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>  <span class="at">-x</span> <span class="st">&quot;./ref_genome/index_hisat2/human&quot;</span> <span class="dt">\</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>  <span class="at">-1</span> <span class="st">&quot;./clean_reads/</span><span class="va">${base}</span><span class="st">_1_clean.fastq.gz&quot;</span> <span class="dt">\</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>  <span class="at">-2</span> <span class="st">&quot;./clean_reads/</span><span class="va">${base}</span><span class="st">_2_clean.fastq.gz&quot;</span> <span class="dt">\</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">-@</span> 12 <span class="at">-o</span> ./reads_mapping/<span class="va">${base}</span>.sorted.bam</span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a>  </span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
</details>
<details>
<summary>
<strong>Manipulate SAM/BAM file with Samtools</strong>
</summary>
<ul>
<li><p>Convert SAM to BAM file (binary format) → save space (done
previously)</p></li>
<li><p>BAM files need to be <em>sorted</em> (done previously) and
<em>indexed</em></p></li>
</ul>
<div class="sourceCode" id="cb7"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># module load samtools</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> reads_mapping/<span class="pp">*</span>.sorted.bam</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="cf">do</span> </span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>  <span class="va">base</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">&quot;</span><span class="va">$sample</span><span class="st">&quot;</span> .sorted.bam<span class="va">)</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>  </span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>  <span class="co"># Index sorted BAM</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>  <span class="ex">samtools</span> index <span class="st">&quot;reads_mapping/</span><span class="va">${base}</span><span class="st">.sorted.bam&quot;</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
</details>
<details>
<summary>
<strong>Generate count table</strong>
</summary>
<p>Now we’ll pass all the sorted, indexed BAM files through
<code>featureCounts</code> function (subread package) to generate the
count matrix. We also need an annotation file to do this. Parameters
<code>-t "exon" -g gene_id</code> are to count reads mapped to exon then
summarize to gene-level (i.e. many exons belong to single gene id).</p>
<div class="sourceCode" id="cb8"><pre
class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co"># Download annotation file (.gtf)</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">wget</span> ftp://ftp.ensembl.org/pub/release-100/gtf/homo_sapiens/Homo_sapiens.GRCh38.100.gtf.gz</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co"># Unzip (then move to ref_genome folder)</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="fu">gunzip</span> Homo_sapiens.GRCh38.100.gtf.gz</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co"># module load subread</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co"># Remember -p to tell featureCounts that BAM files contain paired-end reads</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="ex">featureCounts</span> <span class="at">-a</span> ./ref_genome/Homo_sapiens.GRCh38.100.gtf <span class="dt">\</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>-s 0 <span class="at">-t</span> <span class="st">&quot;exon&quot;</span> <span class="at">-g</span> gene_id <span class="dt">\</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>-p <span class="at">--countReadPairs</span> <span class="dt">\</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>-o ./count_matrix.counts <span class="dt">\</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>./reads_mapping/SRR<span class="pp">*</span>.sorted.bam</span></code></pre></div>
</details>
<p>Now we can read the <strong>count matrix</strong> into R:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="co"># Read count table</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>count_tab <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">file =</span> <span class="st">&quot;./projects/project_1/dmd-organoid_count_matrix.counts&quot;</span>,</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>                        <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>                        <span class="at">row.names =</span> <span class="dv">1</span>,</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>                        <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>                        <span class="at">comment.char =</span> <span class="st">&quot;#&quot;</span>)</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="co"># Clean column names</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a><span class="fu">colnames</span>(count_tab) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^.*</span><span class="sc">\\</span><span class="st">/&quot;</span>, <span class="st">&quot;&quot;</span>, </span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>                         <span class="fu">colnames</span>(count_tab))    <span class="co"># to match everything from the start of the string up to and including the last slash /</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a></span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a><span class="fu">colnames</span>(count_tab) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.sorted</span><span class="sc">\\</span><span class="st">.bam$&quot;</span>, <span class="st">&quot;&quot;</span>, </span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>                         <span class="fu">colnames</span>(count_tab))    <span class="co"># drop &quot;.sorted.bam&quot;</span></span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a></span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a><span class="fu">colnames</span>(count_tab) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;^</span><span class="sc">\\</span><span class="st">.</span><span class="sc">\\</span><span class="st">.reads_mapping</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;&quot;</span>, </span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a>                            <span class="fu">colnames</span>(count_tab)) <span class="co"># from the start, drop &quot;..reads_mapping&quot;</span></span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a></span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a><span class="fu">colnames</span>(count_tab)</span></code></pre></div>
<pre><code>##  [1] &quot;Chr&quot;         &quot;Start&quot;       &quot;End&quot;         &quot;Strand&quot;      &quot;Length&quot;     
##  [6] &quot;SRR32076135&quot; &quot;SRR32076136&quot; &quot;SRR32076137&quot; &quot;SRR32076138&quot; &quot;SRR32076139&quot;
## [11] &quot;SRR32076140&quot; &quot;SRR32076141&quot; &quot;SRR32076142&quot; &quot;SRR32076143&quot; &quot;SRR32076144&quot;
## [16] &quot;SRR32076145&quot; &quot;SRR32076146&quot;</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co"># Drop annotation columns and keep the counts only</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>count_tab <span class="ot">&lt;-</span> count_tab[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)]</span></code></pre></div>
<table class="table table-striped table-hover table-condensed table-responsive" style="margin-left: auto; margin-right: auto;">
<caption>
Count table: genes as rows, samples as columns
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
SRR32076135
</th>
<th style="text-align:right;">
SRR32076136
</th>
<th style="text-align:right;">
SRR32076137
</th>
<th style="text-align:right;">
SRR32076138
</th>
<th style="text-align:right;">
SRR32076139
</th>
<th style="text-align:right;">
SRR32076140
</th>
<th style="text-align:right;">
SRR32076141
</th>
<th style="text-align:right;">
SRR32076142
</th>
<th style="text-align:right;">
SRR32076143
</th>
<th style="text-align:right;">
SRR32076144
</th>
<th style="text-align:right;">
SRR32076145
</th>
<th style="text-align:right;">
SRR32076146
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSG00000186092
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000284733
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000284662
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000187634
</td>
<td style="text-align:right;">
414
</td>
<td style="text-align:right;">
289
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:right;">
437
</td>
<td style="text-align:right;">
242
</td>
<td style="text-align:right;">
375
</td>
<td style="text-align:right;">
337
</td>
<td style="text-align:right;">
138
</td>
<td style="text-align:right;">
419
</td>
<td style="text-align:right;">
345
</td>
<td style="text-align:right;">
117
</td>
<td style="text-align:right;">
111
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000188976
</td>
<td style="text-align:right;">
2342
</td>
<td style="text-align:right;">
2056
</td>
<td style="text-align:right;">
1091
</td>
<td style="text-align:right;">
2672
</td>
<td style="text-align:right;">
1551
</td>
<td style="text-align:right;">
2108
</td>
<td style="text-align:right;">
2394
</td>
<td style="text-align:right;">
3062
</td>
<td style="text-align:right;">
2306
</td>
<td style="text-align:right;">
2449
</td>
<td style="text-align:right;">
1196
</td>
<td style="text-align:right;">
2706
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000187961
</td>
<td style="text-align:right;">
627
</td>
<td style="text-align:right;">
557
</td>
<td style="text-align:right;">
259
</td>
<td style="text-align:right;">
643
</td>
<td style="text-align:right;">
361
</td>
<td style="text-align:right;">
502
</td>
<td style="text-align:right;">
642
</td>
<td style="text-align:right;">
563
</td>
<td style="text-align:right;">
594
</td>
<td style="text-align:right;">
559
</td>
<td style="text-align:right;">
278
</td>
<td style="text-align:right;">
495
</td>
</tr>
</tbody>
</table>
<p>And <strong>metadata</strong> file:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;./projects/project_1/dmd-organoid_metadata.xlsx&quot;</span>)</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(metadata[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)]) <span class="co"># keep 2 columns: sample and sample_type</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>meta<span class="sc">$</span>sample_type <span class="ot">&lt;-</span> <span class="fu">factor</span>(meta<span class="sc">$</span>sample_type) <span class="co"># sample_type as factor</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="co"># sample names have to be rownames</span></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="fu">rownames</span>(meta) <span class="ot">&lt;-</span> meta<span class="sc">$</span>sample</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>meta<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a><span class="co"># Map row names (sample names) from meta to column names of count table</span></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>name_map <span class="ot">&lt;-</span> <span class="fu">setNames</span>(metadata<span class="sc">$</span>sample, metadata<span class="sc">$</span>sample_id)</span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a><span class="fu">colnames</span>(count_tab) <span class="ot">&lt;-</span> name_map[<span class="fu">colnames</span>(count_tab)]</span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a></span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a><span class="co"># Reorder meta to match the order of columns in count table</span></span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a>meta <span class="ot">&lt;-</span> meta[<span class="fu">colnames</span>(count_tab), , drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb12-16"><a href="#cb12-16" tabindex="-1"></a></span>
<span id="cb12-17"><a href="#cb12-17" tabindex="-1"></a><span class="co"># Check if it&#39;s good to go</span></span>
<span id="cb12-18"><a href="#cb12-18" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(meta) <span class="sc">==</span> <span class="fu">colnames</span>(count_tab))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<table class="table table-striped table-hover table-condensed table-responsive" style="margin-left: auto; margin-right: auto;">
<caption>
Sample metadata
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
sample_type
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
DMDdEx8-9-high-1
</td>
<td style="text-align:left;">
Treated
</td>
</tr>
<tr>
<td style="text-align:left;">
DMDdEx8-9-high-2
</td>
<td style="text-align:left;">
Treated
</td>
</tr>
<tr>
<td style="text-align:left;">
corrDMDdEx8-9-4
</td>
<td style="text-align:left;">
IsoCTR
</td>
</tr>
<tr>
<td style="text-align:left;">
DMDdEx8-9-1
</td>
<td style="text-align:left;">
Untreated
</td>
</tr>
<tr>
<td style="text-align:left;">
DMDdEx8-9-2
</td>
<td style="text-align:left;">
Untreated
</td>
</tr>
<tr>
<td style="text-align:left;">
DMDdEx8-9-3
</td>
<td style="text-align:left;">
Untreated
</td>
</tr>
<tr>
<td style="text-align:left;">
DMDdEx8-9-4
</td>
<td style="text-align:left;">
Untreated
</td>
</tr>
<tr>
<td style="text-align:left;">
corrDMDdEx8-9-3
</td>
<td style="text-align:left;">
IsoCTR
</td>
</tr>
<tr>
<td style="text-align:left;">
DMDdEx8-9-high-4
</td>
<td style="text-align:left;">
Treated
</td>
</tr>
<tr>
<td style="text-align:left;">
DMDdEx8-9-high-3
</td>
<td style="text-align:left;">
Treated
</td>
</tr>
<tr>
<td style="text-align:left;">
corrDMDdEx8-9-2
</td>
<td style="text-align:left;">
IsoCTR
</td>
</tr>
<tr>
<td style="text-align:left;">
corrDMDdEx8-9-1
</td>
<td style="text-align:left;">
IsoCTR
</td>
</tr>
</tbody>
</table>
</div>
<div id="differential-expression-analysis-with-deseq2"
class="section level1">
<h1>Differential expression analysis with DESeq2</h1>
<div id="create-deseqdataset-object" class="section level2">
<h2>Create DESeqDataSet object</h2>
<p>DESeq2 expects a raw count matrix as input. The model will internally
correct for library size. Remember that
<code>nrow(colData) == ncol(countData)</code>.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>  <span class="at">countData =</span> count_tab, </span>
<span id="cb14-4"><a href="#cb14-4" tabindex="-1"></a>  <span class="at">colData =</span> meta,</span>
<span id="cb14-5"><a href="#cb14-5" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> sample_type)</span>
<span id="cb14-6"><a href="#cb14-6" tabindex="-1"></a>dds</span></code></pre></div>
<pre><code>## class: DESeqDataSet 
## dim: 20420 12 
## metadata(1): version
## assays(1): counts
## rownames(20420): ENSG00000186092 ENSG00000284733 ... ENSG00000277475
##   ENSG00000268674
## rowData names(0):
## colnames(12): DMDdEx8-9-high-1 DMDdEx8-9-high-2 ... corrDMDdEx8-9-2
##   corrDMDdEx8-9-1
## colData names(1): sample_type</code></pre>
</div>
<div id="filter-low-count-genes" class="section level2">
<h2>Filter low-count genes</h2>
<p>Here we’re keeping only genes that have at least 10 read counts in at
least 4 samples. The mean-sd plot suggests that variance is stabilized,
meaning genes with low and high mean counts all have similar
variances.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>smallestGroupSize <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> smallestGroupSize</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>dds_filt <span class="ot">&lt;-</span> dds[keep,]</span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a><span class="fu">library</span>(vsn)</span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds_filt)</span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a><span class="fu">meanSdPlot</span>(<span class="fu">assay</span>(vsd))</span></code></pre></div>
<p><img src="project_1_dmd-organoid_files/figure-html/gene_filtering-1.png" width="576" /></p>
</div>
<div id="data-quality-control-by-clustering-samples"
class="section level2">
<h2>Data quality control by clustering samples</h2>
<p>For differential gene expression (DEG) analysis, DESeq2 operates
directly on the raw counts. However, for clustering and visualization,
we better use <em>normalized</em> and <em>transformed</em> expression
values. DESeq2 uses a normalization method called <strong>median of
ratios</strong>.</p>
<div class="line-block">Note: We have to run DEG analysis to add
normalization factors (since it’s an all-in-one function)</div>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># DESeqDataSet object does not store a matrix of normalized counts by default</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="co"># We can to get it by: counts(dds, normalized=TRUE)</span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>dds_filt <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds_filt)</span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>select_top_genes <span class="ot">&lt;-</span> <span class="fu">order</span>(<span class="fu">rowMeans</span>(<span class="fu">counts</span>(dds_filt, <span class="at">normalized=</span><span class="cn">TRUE</span>)), <span class="co"># mean of normalized counts of each gene across samples                 </span></span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a>                                     <span class="at">decreasing=</span><span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>] <span class="co"># select top 50 genes with highest mean expression  </span></span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a></span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds_filt)[, <span class="st">&quot;sample_type&quot;</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a><span class="co"># log2 transformation of normalized counts </span></span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a><span class="co"># pc = 1 is pseudocount ==&gt; log2(n + 1) </span></span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a>ntd <span class="ot">&lt;-</span> <span class="fu">normTransform</span>(dds_filt, <span class="at">f =</span> log2, <span class="at">pc =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>This is a heatmap of top 50 highly expressed genes. Expression values
are presented as normalized, log2-transformed counts and are row-scaled
(so that mean expression of each gene across samples is 0). Here we see
samples are clustered based on top gene expression.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;pheatmap&quot;</span>)</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="fu">assay</span>(ntd)[select_top_genes,],          </span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">&quot;row&quot;</span>, <span class="co"># z score     </span></span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>         <span class="at">cluster_rows=</span><span class="cn">TRUE</span>,           </span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>         <span class="at">show_rownames=</span><span class="cn">TRUE</span>,                    </span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>         <span class="at">cluster_cols=</span><span class="cn">TRUE</span>,           </span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a>         <span class="at">annotation_col=</span>df)</span></code></pre></div>
<p><img src="project_1_dmd-organoid_files/figure-html/top_gene_heatmap-1.png" width="768" /></p>
<p>We can make a heatmap of (euclidean) distances between each pair of
samples. With this, we have an overall idea of how samples are similar
to each other.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>sampleDists <span class="ot">&lt;-</span> <span class="fu">dist</span>(<span class="fu">t</span>(<span class="fu">assay</span>(vsd)))  </span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;RColorBrewer&quot;</span>) </span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>sampleDistMatrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(sampleDists)</span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Blues&quot;</span>)))(<span class="dv">255</span>) </span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a><span class="fu">pheatmap</span>(sampleDistMatrix,          </span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>         <span class="at">clustering_distance_rows=</span>sampleDists,    </span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a>         <span class="at">clustering_distance_cols=</span>sampleDists,          </span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a>         <span class="at">col=</span>colors)</span></code></pre></div>
<p><img src="project_1_dmd-organoid_files/figure-html/dist_matrix_heatmap-1.png" width="768" /></p>
<p>Another way to evaluate the batch effects is to plot the principle
components of samples (PCA).</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup=</span><span class="st">&quot;sample_type&quot;</span>, <span class="at">ntop=</span><span class="dv">10000</span>)</span></code></pre></div>
<p><img src="project_1_dmd-organoid_files/figure-html/pca-1.png" width="480" /></p>
</div>
<div id="deg-analysis" class="section level2">
<h2>DEG analysis</h2>
<p><code>DESeq()</code> performs:</p>
<ul>
<li><p>estimation of size factors</p></li>
<li><p>estimation of dispersion</p></li>
<li><p>Negative Binomial GLM fitting and Wald statistics</p></li>
</ul>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co"># If needed, set &#39;IsoCTR&#39; as baseline condition :</span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="co"># dds_filt$sample_type &lt;- relevel(dds_filt$sample_type, ref = &quot;IsoCTR&quot;)</span></span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a><span class="co"># dds_filt &lt;- DESeq(dds_filt)</span></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a><span class="fu">resultsNames</span>(dds_filt) <span class="co"># names of the individual effects (coefficients)</span></span></code></pre></div>
<pre><code>## [1] &quot;Intercept&quot;                       &quot;sample_type_Treated_vs_IsoCTR&quot;  
## [3] &quot;sample_type_Untreated_vs_IsoCTR&quot;</code></pre>
<p>p values are corrected by Benjamin-Hochberg with
<code>alpha = 0.05</code>. <strong>How many genes were (significantly)
up/down-regulated in ‘Treated’ compared to ‘IsoCTR’?</strong></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>Treated_vs_IsoCTR <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_filt,</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>                             <span class="at">name =</span> <span class="st">&quot;sample_type_Treated_vs_IsoCTR&quot;</span>,</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>                             <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a><span class="fu">summary</span>(Treated_vs_IsoCTR)</span></code></pre></div>
<pre><code>## 
## out of 14658 with nonzero total read count
## adjusted p-value &lt; 0.05
## LFC &gt; 0 (up)       : 3771, 26%
## LFC &lt; 0 (down)     : 4262, 29%
## outliers [1]       : 3, 0.02%
## low counts [2]     : 0, 0%
## (mean count &lt; 5)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p>For the volcano plot, we’ll use the full <code>results</code> table
and highlight the significant hits. Be careful that there are genes
where <code>padj</code> is missing due to independent filtering inside
DESeq2 (prior to multiple testing correction) or
<code>log2FoldChange</code> is missing if a gene has zero counts across
all samples of one condition.</p>
<details>
<summary>
<strong>Make a volcano plot of DEGs</strong>
</summary>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="co"># Libraries </span></span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a><span class="co"># Prepare DEGs data frame</span></span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>Treated_vs_IsoCTR_DEGs <span class="ot">&lt;-</span> Treated_vs_IsoCTR <span class="sc">%&gt;%</span> </span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;ENSEMBL&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a>  <span class="co"># Remove genes where log2FC or padj is missing</span></span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(log2FoldChange), <span class="sc">!</span><span class="fu">is.na</span>(padj))</span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a></span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a><span class="co"># Add column specifying Up/Down/NotSig genes</span></span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a>Treated_vs_IsoCTR_DEGs<span class="sc">$</span>regulation <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a>  Treated_vs_IsoCTR_DEGs<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> </span>
<span id="cb25-18"><a href="#cb25-18" tabindex="-1"></a>    Treated_vs_IsoCTR_DEGs<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="st">&quot;Up&quot;</span>,</span>
<span id="cb25-19"><a href="#cb25-19" tabindex="-1"></a>  <span class="fu">ifelse</span>( Treated_vs_IsoCTR_DEGs<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span></span>
<span id="cb25-20"><a href="#cb25-20" tabindex="-1"></a>            Treated_vs_IsoCTR_DEGs<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>, <span class="st">&quot;Down&quot;</span>,</span>
<span id="cb25-21"><a href="#cb25-21" tabindex="-1"></a>          <span class="st">&quot;NotSig&quot;</span>)</span>
<span id="cb25-22"><a href="#cb25-22" tabindex="-1"></a>  )</span>
<span id="cb25-23"><a href="#cb25-23" tabindex="-1"></a>  </span>
<span id="cb25-24"><a href="#cb25-24" tabindex="-1"></a><span class="co"># Count up/down genes</span></span>
<span id="cb25-25"><a href="#cb25-25" tabindex="-1"></a>counts <span class="ot">&lt;-</span> Treated_vs_IsoCTR_DEGs <span class="sc">%&gt;%</span></span>
<span id="cb25-26"><a href="#cb25-26" tabindex="-1"></a>  <span class="fu">filter</span>(regulation <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Up&quot;</span>,<span class="st">&quot;Down&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb25-27"><a href="#cb25-27" tabindex="-1"></a>  <span class="fu">group_by</span>(regulation) <span class="sc">%&gt;%</span></span>
<span id="cb25-28"><a href="#cb25-28" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-29"><a href="#cb25-29" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> regulation, <span class="at">values_from =</span> n, <span class="at">values_fill =</span> <span class="dv">0</span>)</span>
<span id="cb25-30"><a href="#cb25-30" tabindex="-1"></a></span>
<span id="cb25-31"><a href="#cb25-31" tabindex="-1"></a><span class="co"># Volcano plot</span></span>
<span id="cb25-32"><a href="#cb25-32" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> Treated_vs_IsoCTR_DEGs,</span>
<span id="cb25-33"><a href="#cb25-33" tabindex="-1"></a>            <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange,</span>
<span id="cb25-34"><a href="#cb25-34" tabindex="-1"></a>                <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj))) <span class="sc">+</span></span>
<span id="cb25-35"><a href="#cb25-35" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">col =</span> regulation)) <span class="sc">+</span></span>
<span id="cb25-36"><a href="#cb25-36" tabindex="-1"></a>  </span>
<span id="cb25-37"><a href="#cb25-37" tabindex="-1"></a>  <span class="co"># Add threshold lines</span></span>
<span id="cb25-38"><a href="#cb25-38" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), </span>
<span id="cb25-39"><a href="#cb25-39" tabindex="-1"></a>             <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, </span>
<span id="cb25-40"><a href="#cb25-40" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>) <span class="sc">+</span></span>
<span id="cb25-41"><a href="#cb25-41" tabindex="-1"></a>  </span>
<span id="cb25-42"><a href="#cb25-42" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>),</span>
<span id="cb25-43"><a href="#cb25-43" tabindex="-1"></a>           <span class="at">col =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb25-44"><a href="#cb25-44" tabindex="-1"></a>           <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>) <span class="sc">+</span></span>
<span id="cb25-45"><a href="#cb25-45" tabindex="-1"></a>  </span>
<span id="cb25-46"><a href="#cb25-46" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Down&quot;</span><span class="ot">=</span><span class="st">&quot;#00AFBB&quot;</span>,</span>
<span id="cb25-47"><a href="#cb25-47" tabindex="-1"></a>                                <span class="st">&quot;NotSig&quot;</span><span class="ot">=</span><span class="st">&quot;grey&quot;</span>,</span>
<span id="cb25-48"><a href="#cb25-48" tabindex="-1"></a>                                <span class="st">&quot;Up&quot;</span><span class="ot">=</span><span class="st">&quot;#E41A1C&quot;</span>),</span>
<span id="cb25-49"><a href="#cb25-49" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Down&quot;</span><span class="ot">=</span><span class="st">&quot;Downregulated&quot;</span>, </span>
<span id="cb25-50"><a href="#cb25-50" tabindex="-1"></a>                                <span class="st">&quot;NotSig&quot;</span><span class="ot">=</span><span class="st">&quot;Not significant&quot;</span>, </span>
<span id="cb25-51"><a href="#cb25-51" tabindex="-1"></a>                                <span class="st">&quot;Up&quot;</span><span class="ot">=</span><span class="st">&quot;Upregulated&quot;</span>)) <span class="sc">+</span></span>
<span id="cb25-52"><a href="#cb25-52" tabindex="-1"></a>  </span>
<span id="cb25-53"><a href="#cb25-53" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;log2FC&quot;</span>,</span>
<span id="cb25-54"><a href="#cb25-54" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;-log10(padj)&quot;</span>,</span>
<span id="cb25-55"><a href="#cb25-55" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">&quot;Regulation&quot;</span>, </span>
<span id="cb25-56"><a href="#cb25-56" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;Treated vs IsoCTR&quot;</span>) <span class="sc">+</span> </span>
<span id="cb25-57"><a href="#cb25-57" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb25-58"><a href="#cb25-58" tabindex="-1"></a>  </span>
<span id="cb25-59"><a href="#cb25-59" tabindex="-1"></a>  <span class="co"># Annotate number of up/down genes on the plot </span></span>
<span id="cb25-60"><a href="#cb25-60" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> <span class="fu">max</span>(Treated_vs_IsoCTR_DEGs<span class="sc">$</span>log2FoldChange, <span class="at">na.rm=</span><span class="cn">TRUE</span>), </span>
<span id="cb25-61"><a href="#cb25-61" tabindex="-1"></a>           <span class="at">y =</span> <span class="fu">max</span>(<span class="sc">-</span><span class="fu">log10</span>(Treated_vs_IsoCTR_DEGs<span class="sc">$</span>padj), <span class="at">na.rm=</span><span class="cn">TRUE</span>), </span>
<span id="cb25-62"><a href="#cb25-62" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;Up: &quot;</span>, counts<span class="sc">$</span>Up), </span>
<span id="cb25-63"><a href="#cb25-63" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">&quot;#E41A1C&quot;</span>) <span class="sc">+</span></span>
<span id="cb25-64"><a href="#cb25-64" tabindex="-1"></a>  </span>
<span id="cb25-65"><a href="#cb25-65" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> <span class="fu">min</span>(Treated_vs_IsoCTR_DEGs<span class="sc">$</span>log2FoldChange, <span class="at">na.rm=</span><span class="cn">TRUE</span>), </span>
<span id="cb25-66"><a href="#cb25-66" tabindex="-1"></a>           <span class="at">y =</span> <span class="fu">max</span>(<span class="sc">-</span><span class="fu">log10</span>(Treated_vs_IsoCTR_DEGs<span class="sc">$</span>padj), <span class="at">na.rm=</span><span class="cn">TRUE</span>), </span>
<span id="cb25-67"><a href="#cb25-67" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;Down: &quot;</span>, counts<span class="sc">$</span>Down), </span>
<span id="cb25-68"><a href="#cb25-68" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">&quot;#00AFBB&quot;</span>)</span></code></pre></div>
</details>
<p><img src="project_1_dmd-organoid_files/figure-html/display_volcano_plot-1.png" width="672" /></p>
<details>
<summary>
Save DEG list to file
</summary>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="co"># Save DEG list</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a><span class="fu">write.table</span>(Treated_vs_IsoCTR_DEGs,</span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">&quot;./projects/project_1/DEG_List_backup/Treated_vs_IsoCTR_DEGs.txt&quot;</span>,</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</details>
<p><strong>How many genes were (significantly) up/down-regulated in
‘Untreated’ compared to ‘IsoCTR’?</strong></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>Untreated_vs_IsoCTR <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_filt,</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>                               <span class="at">name =</span> <span class="st">&quot;sample_type_Untreated_vs_IsoCTR&quot;</span>,</span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>                               <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a><span class="fu">summary</span>(Untreated_vs_IsoCTR)</span></code></pre></div>
<pre><code>## 
## out of 14658 with nonzero total read count
## adjusted p-value &lt; 0.05
## LFC &gt; 0 (up)       : 1933, 13%
## LFC &lt; 0 (down)     : 2685, 18%
## outliers [1]       : 3, 0.02%
## low counts [2]     : 0, 0%
## (mean count &lt; 5)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p><strong>How many genes were (significantly) up/down-regulated in
‘Treated’ compared to ‘Untreated’?</strong></p>
<pre><code>## 
## out of 14658 with nonzero total read count
## adjusted p-value &lt; 0.05
## LFC &gt; 0 (up)       : 315, 2.1%
## LFC &lt; 0 (down)     : 389, 2.7%
## outliers [1]       : 3, 0.02%
## low counts [2]     : 285, 1.9%
## (mean count &lt; 9)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
</div>
</div>
<div id="crossing-deg-sets-venn-diagram" class="section level1">
<h1>Crossing DEG sets: Venn diagram</h1>
<p>We want to intersect 3 sets of DEGs:</p>
<ul>
<li><p><strong>Untreated vs IsoCTR:</strong> genes that are
differentially expressed in DMD organoid compared to isogenic
control</p></li>
<li><p><strong>Treated vs Untreated:</strong> genes that are
differentially expressed in AAV-treated DMD organoid compared to
non-treated (presumably ‘corrected’ by gene therapy)</p></li>
<li><p><strong>Treated vs IsoCTR:</strong> presumably ‘not corrected’ by
gene therapy</p></li>
</ul>
<div class="line-block">Note: We can use
<code>abs(log2FoldChange) &gt; 1</code> but it seems too strict.</div>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>  <span class="fu">rownames</span>(Untreated_vs_IsoCTR <span class="sc">%&gt;%</span> </span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a>             <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a>             <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">0</span>)),</span>
<span id="cb30-5"><a href="#cb30-5" tabindex="-1"></a>  <span class="fu">rownames</span>(Treated_vs_Untreated <span class="sc">%&gt;%</span> </span>
<span id="cb30-6"><a href="#cb30-6" tabindex="-1"></a>             <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb30-7"><a href="#cb30-7" tabindex="-1"></a>             <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">0</span>)),</span>
<span id="cb30-8"><a href="#cb30-8" tabindex="-1"></a>  <span class="fu">rownames</span>(Treated_vs_IsoCTR <span class="sc">%&gt;%</span> </span>
<span id="cb30-9"><a href="#cb30-9" tabindex="-1"></a>             <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb30-10"><a href="#cb30-10" tabindex="-1"></a>             <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb30-11"><a href="#cb30-11" tabindex="-1"></a>)</span>
<span id="cb30-12"><a href="#cb30-12" tabindex="-1"></a></span>
<span id="cb30-13"><a href="#cb30-13" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggVennDiagram&quot;</span>)</span>
<span id="cb30-14"><a href="#cb30-14" tabindex="-1"></a><span class="fu">ggVennDiagram</span>(x, <span class="at">label_alpha =</span> <span class="dv">0</span>,</span>
<span id="cb30-15"><a href="#cb30-15" tabindex="-1"></a>              <span class="at">label =</span> <span class="st">&quot;count&quot;</span>,</span>
<span id="cb30-16"><a href="#cb30-16" tabindex="-1"></a>              <span class="at">category.names =</span> <span class="fu">c</span>(<span class="st">&quot;Untreated</span><span class="sc">\n</span><span class="st">vs IsoCTR&quot;</span>,</span>
<span id="cb30-17"><a href="#cb30-17" tabindex="-1"></a>                                 <span class="st">&quot;Treated</span><span class="sc">\n</span><span class="st">vs Untreated&quot;</span>,</span>
<span id="cb30-18"><a href="#cb30-18" tabindex="-1"></a>                                 <span class="st">&quot;Treated</span><span class="sc">\n</span><span class="st">vs IsoCTR&quot;</span>)) <span class="sc">+</span></span>
<span id="cb30-19"><a href="#cb30-19" tabindex="-1"></a>  </span>
<span id="cb30-20"><a href="#cb30-20" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">4</span>, <span class="st">&quot;cm&quot;</span>),</span>
<span id="cb30-21"><a href="#cb30-21" tabindex="-1"></a>        <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>)) <span class="sc">+</span></span>
<span id="cb30-22"><a href="#cb30-22" tabindex="-1"></a>  </span>
<span id="cb30-23"><a href="#cb30-23" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;white&quot;</span>, <span class="at">high =</span> <span class="st">&quot;orange&quot;</span>)</span></code></pre></div>
<p><img src="project_1_dmd-organoid_files/figure-html/venn_diagram-1.png" width="960" /></p>
</div>
<div id="gene-set-enrichment-analysis-gsea" class="section level1">
<h1>Gene Set Enrichment Analysis (GSEA)</h1>
<p>Prior to GSEA, we need to create a <strong>ranked list</strong> of
the genes.</p>
<details>
<summary>
Tidy and save Treated_vs_Untreated DEG table
</summary>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="co"># Remove genes where log2FC or padj is missing</span></span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a><span class="co"># Make row names a column EMSEMBL</span></span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a>Treated_vs_Untreated_DEGs <span class="ot">&lt;-</span> </span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a>  Treated_vs_Untreated <span class="sc">%&gt;%</span></span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(log2FoldChange), <span class="sc">!</span><span class="fu">is.na</span>(padj)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;ENSEMBL&quot;</span>)</span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a></span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a><span class="co"># Add column &#39;regulation&#39;</span></span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a>Treated_vs_Untreated_DEGs<span class="sc">$</span>regulation <span class="ot">&lt;-</span></span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a><span class="fu">ifelse</span>( Treated_vs_Untreated_DEGs<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span></span>
<span id="cb31-13"><a href="#cb31-13" tabindex="-1"></a>          Treated_vs_Untreated_DEGs<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="st">&quot;Up&quot;</span>,</span>
<span id="cb31-14"><a href="#cb31-14" tabindex="-1"></a>  <span class="fu">ifelse</span>( Treated_vs_Untreated_DEGs<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span></span>
<span id="cb31-15"><a href="#cb31-15" tabindex="-1"></a>            Treated_vs_Untreated_DEGs<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>, <span class="st">&quot;Down&quot;</span>,</span>
<span id="cb31-16"><a href="#cb31-16" tabindex="-1"></a>          <span class="st">&quot;NotSig&quot;</span>)</span>
<span id="cb31-17"><a href="#cb31-17" tabindex="-1"></a>)</span>
<span id="cb31-18"><a href="#cb31-18" tabindex="-1"></a></span>
<span id="cb31-19"><a href="#cb31-19" tabindex="-1"></a><span class="co"># Write the file</span></span>
<span id="cb31-20"><a href="#cb31-20" tabindex="-1"></a><span class="fu">write.table</span>(Treated_vs_Untreated_DEGs,</span>
<span id="cb31-21"><a href="#cb31-21" tabindex="-1"></a>            <span class="at">file =</span> <span class="st">&quot;./projects/project_1/DEG_List_backup/Treated_vs_Untreated_DEGs.txt&quot;</span>,</span>
<span id="cb31-22"><a href="#cb31-22" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb31-23"><a href="#cb31-23" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb31-24"><a href="#cb31-24" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</details>
<p>A bunch of ranking metrics can be used to rank the list of input
genes. In the end, they should all correlate to the two phenotypes
signatured by the gene set that we’re interested in.</p>
<p>Here we’ll try using the <code>log2FoldChange</code> in DESeq2.</p>
<div class="line-block">Using shrinkage of LFC estimates (toward 0) is
to avoid inflation by genes with small counts/large standard
errors.</div>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a>Treat_vs_Untreated_DEGs <span class="ot">&lt;-</span></span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a><span class="fu">read.table</span>(<span class="at">file =</span> <span class="st">&quot;./projects/project_1/DEG_List/Treated_vs_Untreated_DEGs.txt&quot;</span>,</span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>           <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a>Treat_vs_Untreated_DEGs_ranked <span class="ot">&lt;-</span> Treat_vs_Untreated_DEGs <span class="sc">%&gt;%</span> </span>
<span id="cb32-6"><a href="#cb32-6" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(log2FoldChange)) <span class="sc">%&gt;%</span></span>
<span id="cb32-7"><a href="#cb32-7" tabindex="-1"></a>  <span class="fu">pull</span>(log2FoldChange, <span class="at">name=</span>ENSEMBL) <span class="co"># create named vector</span></span>
<span id="cb32-8"><a href="#cb32-8" tabindex="-1"></a></span>
<span id="cb32-9"><a href="#cb32-9" tabindex="-1"></a><span class="fu">head</span>(Treat_vs_Untreated_DEGs_ranked)</span></code></pre></div>
<pre><code>## ENSG00000166450 ENSG00000169840 ENSG00000105810 ENSG00000274276 ENSG00000187553 
##        1.739995        1.496896        1.495937        1.470676        1.470314 
## ENSG00000163273 
##        1.364892</code></pre>
<p>The <strong>gene set</strong> of interest can be obtained via the <a
href="https://www.gsea-msigdb.org/gsea/msigdb">Molecular Signatures
Database (MSigDB)</a>.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>GOBP_all <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(</span>
<span id="cb34-3"><a href="#cb34-3" tabindex="-1"></a>  <span class="at">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>,</span>
<span id="cb34-4"><a href="#cb34-4" tabindex="-1"></a>  <span class="at">collection =</span> <span class="st">&quot;C5&quot;</span>,</span>
<span id="cb34-5"><a href="#cb34-5" tabindex="-1"></a>  <span class="at">subcollection =</span> <span class="st">&quot;GO:BP&quot;</span></span>
<span id="cb34-6"><a href="#cb34-6" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-7"><a href="#cb34-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gs_name, ensembl_gene)</span>
<span id="cb34-8"><a href="#cb34-8" tabindex="-1"></a><span class="fu">head</span>(GOBP_all)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 2
##   gs_name                                          ensembl_gene   
##   &lt;chr&gt;                                            &lt;chr&gt;          
## 1 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS ENSG00000149313
## 2 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS ENSG00000144908
## 3 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS ENSG00000136010
## 4 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS ENSG00000100714
## 5 GOBP_10_FORMYLTETRAHYDROFOLATE_METABOLIC_PROCESS ENSG00000120254
## 6 GOBP_2FE_2S_CLUSTER_ASSEMBLY                     ENSG00000183336</code></pre>
<p>Running GSEA with all GOBP gene sets:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a>gsea_GOBP_all <span class="ot">&lt;-</span> <span class="fu">GSEA</span>(</span>
<span id="cb36-3"><a href="#cb36-3" tabindex="-1"></a>  Treat_vs_Untreated_DEGs_ranked,</span>
<span id="cb36-4"><a href="#cb36-4" tabindex="-1"></a>  <span class="at">TERM2GENE =</span> GOBP_all,</span>
<span id="cb36-5"><a href="#cb36-5" tabindex="-1"></a>  <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb36-6"><a href="#cb36-6" tabindex="-1"></a></span>
<span id="cb36-7"><a href="#cb36-7" tabindex="-1"></a><span class="co"># List of gene sets with significant enrichment</span></span>
<span id="cb36-8"><a href="#cb36-8" tabindex="-1"></a><span class="fu">head</span>(gsea_GOBP_all<span class="sc">@</span>result, <span class="dv">10</span>)<span class="sc">$</span>ID</span></code></pre></div>
<pre><code>##  [1] &quot;GOBP_MITOTIC_SISTER_CHROMATID_SEGREGATION&quot;              
##  [2] &quot;GOBP_SISTER_CHROMATID_SEGREGATION&quot;                      
##  [3] &quot;GOBP_NEGATIVE_REGULATION_OF_CHROMOSOME_ORGANIZATION&quot;    
##  [4] &quot;GOBP_MITOTIC_CELL_CYCLE_CHECKPOINT_SIGNALING&quot;           
##  [5] &quot;GOBP_DEFENSE_RESPONSE_TO_BACTERIUM&quot;                     
##  [6] &quot;GOBP_REGULATION_OF_CHROMOSOME_SEGREGATION&quot;              
##  [7] &quot;GOBP_REGULATION_OF_MITOTIC_SISTER_CHROMATID_SEGREGATION&quot;
##  [8] &quot;GOBP_CHROMOSOME_SEGREGATION&quot;                            
##  [9] &quot;GOBP_CHROMOSOME_SEPARATION&quot;                             
## [10] &quot;GOBP_NUCLEAR_CHROMOSOME_SEGREGATION&quot;</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb38-2"><a href="#cb38-2" tabindex="-1"></a><span class="fu">dotplot</span>(gsea_GOBP_all, <span class="at">showCategory =</span> <span class="dv">20</span>,</span>
<span id="cb38-3"><a href="#cb38-3" tabindex="-1"></a>        <span class="at">color =</span> <span class="st">&quot;NES&quot;</span>) <span class="sc">+</span></span>
<span id="cb38-4"><a href="#cb38-4" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Top 20 enriched pathways&quot;</span>)</span></code></pre></div>
<p><img src="project_1_dmd-organoid_files/figure-html/dotplot-1.png" width="768" /></p>
<details>
<summary>
<strong>How to see only terms that relate to “inflammation”?</strong>
</summary>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="co"># Subset result table for terms containing &quot;*inflam*&quot;</span></span>
<span id="cb39-2"><a href="#cb39-2" tabindex="-1"></a>inflam_terms <span class="ot">&lt;-</span> gsea_GOBP_all<span class="sc">@</span>result <span class="sc">%&gt;%</span></span>
<span id="cb39-3"><a href="#cb39-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;inflam&quot;</span>, Description, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>))</span>
<span id="cb39-4"><a href="#cb39-4" tabindex="-1"></a></span>
<span id="cb39-5"><a href="#cb39-5" tabindex="-1"></a><span class="co"># Create a new gseaResult object</span></span>
<span id="cb39-6"><a href="#cb39-6" tabindex="-1"></a>gsea_GOBP_inflam <span class="ot">&lt;-</span> gsea_GOBP_all</span>
<span id="cb39-7"><a href="#cb39-7" tabindex="-1"></a><span class="co"># The result table in this new object is the subset one</span></span>
<span id="cb39-8"><a href="#cb39-8" tabindex="-1"></a>gsea_GOBP_inflam<span class="sc">@</span>result <span class="ot">&lt;-</span> inflam_terms</span>
<span id="cb39-9"><a href="#cb39-9" tabindex="-1"></a></span>
<span id="cb39-10"><a href="#cb39-10" tabindex="-1"></a><span class="co"># Plot only &#39;*inflam*&#39;-related sets</span></span>
<span id="cb39-11"><a href="#cb39-11" tabindex="-1"></a><span class="fu">dotplot</span>(gsea_GOBP_inflam, <span class="at">color =</span> <span class="st">&quot;NES&quot;</span>,</span>
<span id="cb39-12"><a href="#cb39-12" tabindex="-1"></a>        <span class="at">showCategory =</span> <span class="fu">nrow</span>(inflam_terms)) <span class="sc">+</span></span>
<span id="cb39-13"><a href="#cb39-13" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Inflammation-related GO terms&quot;</span>)</span></code></pre></div>
<p><img src="project_1_dmd-organoid_files/figure-html/inflamm_GOBP-1.png" width="768" /></p>
</details>
<p>Plot enrichment score for terms/signatures that we’re particularly
interested in:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a><span class="co"># Visualization</span></span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb40-3"><a href="#cb40-3" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">gseaplot2</span>(gsea_GOBP_all, <span class="at">geneSetID =</span> <span class="st">&quot;GOBP_POSITIVE_REGULATION_OF_INFLAMMATORY_RESPONSE&quot;</span>,</span>
<span id="cb40-4"><a href="#cb40-4" tabindex="-1"></a>                            <span class="at">pvalue_table =</span> <span class="cn">TRUE</span>)</span>
<span id="cb40-5"><a href="#cb40-5" tabindex="-1"></a>p1_list <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(p1, <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb40-6"><a href="#cb40-6" tabindex="-1"></a>                      <span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.2</span>,<span class="fl">0.5</span>),</span>
<span id="cb40-7"><a href="#cb40-7" tabindex="-1"></a>                      <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span>
<span id="cb40-8"><a href="#cb40-8" tabindex="-1"></a></span>
<span id="cb40-9"><a href="#cb40-9" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">gseaplot2</span>(gsea_GOBP_all, <span class="at">geneSetID =</span> <span class="st">&quot;GOBP_MICROTUBULE_ORGANIZING_CENTER_ORGANIZATION&quot;</span>,</span>
<span id="cb40-10"><a href="#cb40-10" tabindex="-1"></a>                            <span class="at">pvalue_table =</span> <span class="cn">TRUE</span>)</span>
<span id="cb40-11"><a href="#cb40-11" tabindex="-1"></a>p2_list <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(p2, <span class="at">ncol =</span> <span class="dv">1</span>,</span>
<span id="cb40-12"><a href="#cb40-12" tabindex="-1"></a>                      <span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">0.2</span>,<span class="fl">0.5</span>),</span>
<span id="cb40-13"><a href="#cb40-13" tabindex="-1"></a>                      <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span>
<span id="cb40-14"><a href="#cb40-14" tabindex="-1"></a></span>
<span id="cb40-15"><a href="#cb40-15" tabindex="-1"></a><span class="co"># Combine 2 plots</span></span>
<span id="cb40-16"><a href="#cb40-16" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb40-17"><a href="#cb40-17" tabindex="-1"></a><span class="fu">plot_grid</span>(p1_list, p2_list, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="project_1_dmd-organoid_files/figure-html/gsea_plot-1.png" width="576" /></p>
</div>
<div id="convert-ensembl-id-to-gene-symbol-optional"
class="section level1">
<h1>Convert ENSEMBL ID to gene symbol (optional)</h1>
<p>There are many ways to get gene names based on Ensembl IDs. However,
some IDs, usually corresponding to non-coding RNAs, do not match a gene
symbol (from <a href="https://www.genenames.org/">HGNC</a> database). If
we care only about protein-coding genes, maybe go back to
<code>featureCounts</code> and consider reads that map CDS.</p>
<ul>
<li><p>Here we’ll try keeping the IDs as they are for those that don’t
have a gene symbol.</p></li>
<li><p>Same for IDs that match multiple gene symbols, we keep the
Ensembl IDs to make it less ambiguous.</p></li>
</ul>
<details>
<summary>
<strong>Convert EMSEMBL ID to gene symbol</strong>
</summary>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb41-2"><a href="#cb41-2" tabindex="-1"></a>annots <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(org.Hs.eg.db,</span>
<span id="cb41-3"><a href="#cb41-3" tabindex="-1"></a>                 </span>
<span id="cb41-4"><a href="#cb41-4" tabindex="-1"></a>                 <span class="co"># ensembl ids in count table</span></span>
<span id="cb41-5"><a href="#cb41-5" tabindex="-1"></a>                 <span class="at">keys=</span><span class="fu">rownames</span>(count_tab),</span>
<span id="cb41-6"><a href="#cb41-6" tabindex="-1"></a>                 </span>
<span id="cb41-7"><a href="#cb41-7" tabindex="-1"></a>                 <span class="at">columns=</span><span class="fu">c</span>(<span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;GENENAME&quot;</span>), </span>
<span id="cb41-8"><a href="#cb41-8" tabindex="-1"></a>                 <span class="at">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>)</span>
<span id="cb41-9"><a href="#cb41-9" tabindex="-1"></a></span>
<span id="cb41-10"><a href="#cb41-10" tabindex="-1"></a><span class="co"># Replace missing SYMBOL with the ENSEMBL ID</span></span>
<span id="cb41-11"><a href="#cb41-11" tabindex="-1"></a>annots<span class="sc">$</span>SYMBOL[<span class="fu">is.na</span>(annots<span class="sc">$</span>SYMBOL)] <span class="ot">&lt;-</span> annots<span class="sc">$</span>ENSEMBL</span>
<span id="cb41-12"><a href="#cb41-12" tabindex="-1"></a></span>
<span id="cb41-13"><a href="#cb41-13" tabindex="-1"></a><span class="co"># Collapse SYMBOL, but if &gt;1 SYMBOL, use the ENSEMBL ID instead</span></span>
<span id="cb41-14"><a href="#cb41-14" tabindex="-1"></a>annots <span class="ot">&lt;-</span> annots <span class="sc">%&gt;%</span></span>
<span id="cb41-15"><a href="#cb41-15" tabindex="-1"></a>  <span class="fu">group_by</span>(ENSEMBL) <span class="sc">%&gt;%</span></span>
<span id="cb41-16"><a href="#cb41-16" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb41-17"><a href="#cb41-17" tabindex="-1"></a>    <span class="at">SYMBOL =</span> <span class="fu">ifelse</span>(<span class="fu">n_distinct</span>(SYMBOL) <span class="sc">==</span> <span class="dv">1</span>,</span>
<span id="cb41-18"><a href="#cb41-18" tabindex="-1"></a>                    <span class="fu">unique</span>(SYMBOL),</span>
<span id="cb41-19"><a href="#cb41-19" tabindex="-1"></a>                    ENSEMBL)</span>
<span id="cb41-20"><a href="#cb41-20" tabindex="-1"></a>  )</span>
<span id="cb41-21"><a href="#cb41-21" tabindex="-1"></a></span>
<span id="cb41-22"><a href="#cb41-22" tabindex="-1"></a><span class="co"># Add &#39;SYMBOL&#39; column to DEG table</span></span>
<span id="cb41-23"><a href="#cb41-23" tabindex="-1"></a>Treated_vs_Untreated_DEGs_annot <span class="ot">&lt;-</span> Treated_vs_Untreated_DEGs <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(annots, <span class="at">by =</span> <span class="st">&quot;ENSEMBL&quot;</span>)</span></code></pre></div>
</details>
<table class="table table-striped table-hover table-condensed table-responsive" style="margin-left: auto; margin-right: auto;">
<caption>
Treated_vs_Untreated DEG table with gene symbols
</caption>
<thead>
<tr>
<th style="text-align:left;">
ENSEMBL
</th>
<th style="text-align:right;">
baseMean
</th>
<th style="text-align:right;">
log2FoldChange
</th>
<th style="text-align:right;">
lfcSE
</th>
<th style="text-align:right;">
stat
</th>
<th style="text-align:right;">
pvalue
</th>
<th style="text-align:right;">
padj
</th>
<th style="text-align:left;">
regulation
</th>
<th style="text-align:left;">
SYMBOL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ENSG00000187634
</td>
<td style="text-align:right;">
262.6958
</td>
<td style="text-align:right;">
-0.0298865
</td>
<td style="text-align:right;">
0.2634260
</td>
<td style="text-align:right;">
-0.1134531
</td>
<td style="text-align:right;">
0.9096714
</td>
<td style="text-align:right;">
0.9665048
</td>
<td style="text-align:left;">
NotSig
</td>
<td style="text-align:left;">
SAMD11
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000188976
</td>
<td style="text-align:right;">
2060.0851
</td>
<td style="text-align:right;">
-0.0382775
</td>
<td style="text-align:right;">
0.1034109
</td>
<td style="text-align:right;">
-0.3701496
</td>
<td style="text-align:right;">
0.7112710
</td>
<td style="text-align:right;">
0.8819458
</td>
<td style="text-align:left;">
NotSig
</td>
<td style="text-align:left;">
NOC2L
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000187961
</td>
<td style="text-align:right;">
484.3473
</td>
<td style="text-align:right;">
0.0274517
</td>
<td style="text-align:right;">
0.1223824
</td>
<td style="text-align:right;">
0.2243111
</td>
<td style="text-align:right;">
0.8225153
</td>
<td style="text-align:right;">
0.9309065
</td>
<td style="text-align:left;">
NotSig
</td>
<td style="text-align:left;">
KLHL17
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000188290
</td>
<td style="text-align:right;">
1008.3230
</td>
<td style="text-align:right;">
-0.2990822
</td>
<td style="text-align:right;">
0.2779788
</td>
<td style="text-align:right;">
-1.0759173
</td>
<td style="text-align:right;">
0.2819643
</td>
<td style="text-align:right;">
0.6450485
</td>
<td style="text-align:left;">
NotSig
</td>
<td style="text-align:left;">
HES4
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000187608
</td>
<td style="text-align:right;">
255.1310
</td>
<td style="text-align:right;">
-1.5477580
</td>
<td style="text-align:right;">
0.4053978
</td>
<td style="text-align:right;">
-3.8178745
</td>
<td style="text-align:right;">
0.0001346
</td>
<td style="text-align:right;">
0.0073547
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:left;">
ISG15
</td>
</tr>
<tr>
<td style="text-align:left;">
ENSG00000188157
</td>
<td style="text-align:right;">
7175.7911
</td>
<td style="text-align:right;">
0.0467966
</td>
<td style="text-align:right;">
0.3369123
</td>
<td style="text-align:right;">
0.1388986
</td>
<td style="text-align:right;">
0.8895303
</td>
<td style="text-align:right;">
0.9592852
</td>
<td style="text-align:left;">
NotSig
</td>
<td style="text-align:left;">
AGRN
</td>
</tr>
</tbody>
</table>
<p><strong>Tutorials</strong></p>
<p><a
href="https://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">Analyzing
RNA-seq data with DESeq2</a></p>
<p><strong>References</strong></p>
<div id="refs" class="references csl-bib-body hanging-indent"
entry-spacing="0">
<div id="ref-palmieri2023a" class="csl-entry">
Palmieri, Laura, Louna Pili, Abbass Jaber, Ai Vu Hong, Matteo Marcello,
Riyad El-Khoury, Guy Brochier, et al. 2023. <span>“Disease Exacerbation
by Fibroblast Inclusion in Duchenne Muscular Dystrophy MYOrganoids
Reveals Limitations of Microdystrophin Therapeutic Efficacy.”</span> <a
href="http://dx.doi.org/10.1101/2023.07.26.550063">http://dx.doi.org/10.1101/2023.07.26.550063</a>.
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
